diff --git a/filament/backend/test/Arguments.cpp b/filament/backend/test/Arguments.cpp
index 49fa4b7..1332d78 100644
--- a/filament/backend/test/Arguments.cpp
+++ b/filament/backend/test/Arguments.cpp
@@ -16,7 +16,7 @@
 
 #include "PlatformRunner.h"
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <iostream>
 #include <string>
diff --git a/filament/backend/test/ShaderGenerator.cpp b/filament/backend/test/ShaderGenerator.cpp
index 8e32948..df67bee 100644
--- a/filament/backend/test/ShaderGenerator.cpp
+++ b/filament/backend/test/ShaderGenerator.cpp
@@ -16,10 +16,10 @@
 
 #include "ShaderGenerator.h"
 
-#include <GlslangToSpv.h>
+#include <glslang/SPIRV/GlslangToSpv.h>
 #include <SPVRemapper.h>
 
-#include <spirv_glsl.hpp>
+#include <spirv_cross/spirv_glsl.hpp>
 #include <spirv_msl.hpp>
 
 #include "../src/GLSLPostProcessor.h"
diff --git a/libs/filamat/src/GLSLPostProcessor.cpp b/libs/filamat/src/GLSLPostProcessor.cpp
index a538d02..850def4 100644
--- a/libs/filamat/src/GLSLPostProcessor.cpp
+++ b/libs/filamat/src/GLSLPostProcessor.cpp
@@ -16,11 +16,11 @@
 
 #include "GLSLPostProcessor.h"
 
-#include <GlslangToSpv.h>
-#include <SPVRemapper.h>
+#include <glslang/SPIRV/GlslangToSpv.h>
+#include <glslang/SPIRV/SPVRemapper.h>
 
-#include <spirv_glsl.hpp>
-#include <spirv_msl.hpp>
+#include <spirv_cross/spirv_glsl.hpp>
+#include <spirv_cross/spirv_msl.hpp>
 
 #include "sca/builtinResource.h"
 #include "sca/GLSLTools.h"
@@ -240,7 +240,7 @@ void GLSLPostProcessor::spirvToMsl(const SpirvBlob *spirv, std::string *outMsl,
         // buffer itself;
         MSLResourceBinding argBufferBinding;
         // the baseType doesn't matter, but can't be UNKNOWN
-        argBufferBinding.basetype = SPIRType::BaseType::Float;
+        //argBufferBinding.basetype = SPIRType::BaseType::Float;
         argBufferBinding.stage = executionModel;
         argBufferBinding.desc_set = bindingPoint + 1;
         argBufferBinding.binding = kArgumentBufferBinding;
@@ -255,7 +255,7 @@ void GLSLPostProcessor::spirvToMsl(const SpirvBlob *spirv, std::string *outMsl,
         auto set = mslCompiler.get_decoration(resource.id, spv::DecorationDescriptorSet);
         auto binding = mslCompiler.get_decoration(resource.id, spv::DecorationBinding);
         MSLResourceBinding newBinding;
-        newBinding.basetype = SPIRType::BaseType::Void;
+        //newBinding.basetype = SPIRType::BaseType::Void;
         newBinding.stage = executionModel;
         newBinding.desc_set = set;
         newBinding.binding = binding;
@@ -542,7 +542,7 @@ void GLSLPostProcessor::fullOptimization(const TShader& tShader,
 
         if (tShader.getStage() == EShLangFragment && glslOptions.es) {
             for (auto i : config.glsl.subpassInputToColorLocation) {
-                glslCompiler.remap_ext_framebuffer_fetch(i.first, i.second, true);
+                glslCompiler.remap_ext_framebuffer_fetch(i.first, i.second);
             }
         }
 
diff --git a/libs/filamat/src/GLSLPostProcessor.h b/libs/filamat/src/GLSLPostProcessor.h
index cb95079..2f3c84e 100644
--- a/libs/filamat/src/GLSLPostProcessor.h
+++ b/libs/filamat/src/GLSLPostProcessor.h
@@ -25,7 +25,7 @@
 
 #include <spirv-tools/optimizer.hpp>
 
-#include <ShaderLang.h>
+#include <glslang/Public/ShaderLang.h>
 
 #include <backend/DriverEnums.h>
 
diff --git a/libs/filamat/src/sca/ASTHelpers.cpp b/libs/filamat/src/sca/ASTHelpers.cpp
index 0ef9ec6..efdaf86 100644
--- a/libs/filamat/src/sca/ASTHelpers.cpp
+++ b/libs/filamat/src/sca/ASTHelpers.cpp
@@ -18,8 +18,8 @@
 
 #include "GLSLTools.h"
 
-#include <intermediate.h>
-#include <localintermediate.h>
+#include <glslang/Include/intermediate.h>
+#include <glslang/MachineIndependent/localintermediate.h>
 
 #include <utils/Log.h>
 
diff --git a/libs/filamat/src/sca/ASTHelpers.h b/libs/filamat/src/sca/ASTHelpers.h
index b4e4983..e297a3a 100644
--- a/libs/filamat/src/sca/ASTHelpers.h
+++ b/libs/filamat/src/sca/ASTHelpers.h
@@ -20,7 +20,7 @@
 #include <deque>
 #include <string>
 #include <vector>
-#include <intermediate.h>
+#include <glslang/Include/intermediate.h>
 
 #include "GLSLTools.h"
 
diff --git a/libs/filamat/src/sca/GLSLTools.cpp b/libs/filamat/src/sca/GLSLTools.cpp
index 0ee77a5..9c59a69 100644
--- a/libs/filamat/src/sca/GLSLTools.cpp
+++ b/libs/filamat/src/sca/GLSLTools.cpp
@@ -27,8 +27,8 @@
 #include "ASTHelpers.h"
 
 // GLSLANG headers
-#include <InfoSink.h>
-#include <localintermediate.h>
+#include <glslang/include/InfoSink.h>
+#include <glslang/MachineIndependent/localintermediate.h>
 
 #include "builtinResource.h"
 
diff --git a/libs/filamat/src/sca/GLSLTools.h b/libs/filamat/src/sca/GLSLTools.h
index 082a684..8d40a8d 100644
--- a/libs/filamat/src/sca/GLSLTools.h
+++ b/libs/filamat/src/sca/GLSLTools.h
@@ -24,7 +24,7 @@
 
 #include <filamat/MaterialBuilder.h>
 
-#include <ShaderLang.h>
+#include <glslang/Public/ShaderLang.h>
 
 class TIntermNode;
 namespace glslang {
diff --git a/libs/geometry/src/MikktspaceImpl.cpp b/libs/geometry/src/MikktspaceImpl.cpp
index f40da14..cb2b914 100644
--- a/libs/geometry/src/MikktspaceImpl.cpp
+++ b/libs/geometry/src/MikktspaceImpl.cpp
@@ -21,7 +21,7 @@
 
 
 #include <meshoptimizer.h>
-#include <mikktspace/mikktspace.h>
+#include <mikktspace.h>
 
 #include <vector>
 
diff --git a/libs/imageio/src/BasisEncoder.cpp b/libs/imageio/src/BasisEncoder.cpp
index 0c24e71..19d7e21 100644
--- a/libs/imageio/src/BasisEncoder.cpp
+++ b/libs/imageio/src/BasisEncoder.cpp
@@ -22,7 +22,7 @@
 
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Warray-bounds"
-#include <basisu_comp.h>
+#include <encoder/basisu_comp.h>
 #pragma clang diagnostic pop
 
 namespace image {
diff --git a/libs/ktxreader/src/Ktx2Reader.cpp b/libs/ktxreader/src/Ktx2Reader.cpp
index 5d53861..d401133 100644
--- a/libs/ktxreader/src/Ktx2Reader.cpp
+++ b/libs/ktxreader/src/Ktx2Reader.cpp
@@ -26,7 +26,7 @@
 
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Warray-bounds"
-#include <basisu_transcoder.h>
+#include <transcoder/basisu_transcoder.h>
 #pragma clang diagnostic pop
 
 using namespace basist;
diff --git a/libs/matdbg/src/DebugServer.cpp b/libs/matdbg/src/DebugServer.cpp
index 95c5a63..fd2caef 100644
--- a/libs/matdbg/src/DebugServer.cpp
+++ b/libs/matdbg/src/DebugServer.cpp
@@ -22,7 +22,7 @@
 #include <utils/Hash.h>
 #include <utils/Log.h>
 
-#include <spirv_glsl.hpp>
+#include <spirv_cross/spirv_glsl.hpp>
 #include <spirv-tools/libspirv.h>
 
 #include <matdbg/JsonWriter.h>
diff --git a/libs/matdbg/src/ShaderExtractor.cpp b/libs/matdbg/src/ShaderExtractor.cpp
index 3f10b60..bee32dc 100644
--- a/libs/matdbg/src/ShaderExtractor.cpp
+++ b/libs/matdbg/src/ShaderExtractor.cpp
@@ -26,7 +26,7 @@
 
 #include <backend/DriverEnums.h>
 
-#include <spirv_glsl.hpp>
+#include <spirv_cross/spirv_glsl.hpp>
 #include <spirv-tools/libspirv.h>
 
 using namespace filament;
diff --git a/libs/matdbg/src/ShaderReplacer.cpp b/libs/matdbg/src/ShaderReplacer.cpp
index 8eb3497..8be176e 100644
--- a/libs/matdbg/src/ShaderReplacer.cpp
+++ b/libs/matdbg/src/ShaderReplacer.cpp
@@ -27,7 +27,7 @@
 
 #include <sstream>
 
-#include <GlslangToSpv.h>
+#include <glslang/SPIRV/GlslangToSpv.h>
 
 #include "sca/builtinResource.h"
 #include "sca/GLSLTools.h"
diff --git a/libs/viewer/src/RemoteServer.cpp b/libs/viewer/src/RemoteServer.cpp
index 441c730..89c492f 100644
--- a/libs/viewer/src/RemoteServer.cpp
+++ b/libs/viewer/src/RemoteServer.cpp
@@ -208,8 +208,8 @@ bool MessageReceiver::handleData(CivetServer* server, struct mg_connection* conn
 
 void MessageSender::sendMessage(const char* label, const char* buffer, size_t bufsize) {
     for (auto iter : connections) {
-        mg_websocket_write(iter.first, 0x80, label, strlen(label) + 1);
-        mg_websocket_write(iter.first, 0x80, buffer, bufsize);
+        mg_websocket_write(const_cast<mg_connection*>(iter.first), 0x80, label, strlen(label) + 1);
+        mg_websocket_write(const_cast<mg_connection*>(iter.first), 0x80, buffer, bufsize);
     }
 }
 
diff --git a/samples/frame_generator.cpp b/samples/frame_generator.cpp
index a6a2164..3fb7e80 100644
--- a/samples/frame_generator.cpp
+++ b/samples/frame_generator.cpp
@@ -25,7 +25,7 @@
 #include <utility>
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/samples/gltf_instances.cpp b/samples/gltf_instances.cpp
index 688272d..5210ebd 100644
--- a/samples/gltf_instances.cpp
+++ b/samples/gltf_instances.cpp
@@ -33,7 +33,7 @@
 
 #include <camutils/Manipulator.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/NameComponentManager.h>
 
diff --git a/samples/gltf_viewer.cpp b/samples/gltf_viewer.cpp
index 6f4bf72..787a459 100644
--- a/samples/gltf_viewer.cpp
+++ b/samples/gltf_viewer.cpp
@@ -41,7 +41,7 @@
 
 #include <camutils/Manipulator.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/NameComponentManager.h>
 
diff --git a/samples/heightfield.cpp b/samples/heightfield.cpp
index 855dabf..92c98ad 100644
--- a/samples/heightfield.cpp
+++ b/samples/heightfield.cpp
@@ -32,7 +32,7 @@
 #include <utils/JobSystem.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <filamentapp/FilamentApp.h>
 
diff --git a/samples/hellotriangle.cpp b/samples/hellotriangle.cpp
index a36df69..590e35a 100644
--- a/samples/hellotriangle.cpp
+++ b/samples/hellotriangle.cpp
@@ -31,7 +31,7 @@
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <cmath>
 #include <iostream>
diff --git a/samples/image_viewer.cpp b/samples/image_viewer.cpp
index d380aae..c3266fc 100644
--- a/samples/image_viewer.cpp
+++ b/samples/image_viewer.cpp
@@ -36,7 +36,7 @@
 
 #include <camutils/Manipulator.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <math/half.h>
 #include <math/vec3.h>
diff --git a/samples/lightbulb.cpp b/samples/lightbulb.cpp
index 5c5d735..385a5ac 100644
--- a/samples/lightbulb.cpp
+++ b/samples/lightbulb.cpp
@@ -19,7 +19,7 @@
 #include <map>
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/EntityManager.h>
 #include <utils/Path.h>
diff --git a/samples/material_sandbox.cpp b/samples/material_sandbox.cpp
index 7778246..2c8e881 100644
--- a/samples/material_sandbox.cpp
+++ b/samples/material_sandbox.cpp
@@ -19,7 +19,7 @@
 #include <map>
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <imgui.h>
 
diff --git a/samples/rendertarget.cpp b/samples/rendertarget.cpp
index 9ff53cc..8cf3de9 100644
--- a/samples/rendertarget.cpp
+++ b/samples/rendertarget.cpp
@@ -35,7 +35,7 @@
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <iostream>
 
diff --git a/samples/sample_cloth.cpp b/samples/sample_cloth.cpp
index b7a5627..ec32060 100644
--- a/samples/sample_cloth.cpp
+++ b/samples/sample_cloth.cpp
@@ -19,7 +19,7 @@
 #include <map>
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/samples/sample_full_pbr.cpp b/samples/sample_full_pbr.cpp
index 77c08ec..edc284f 100644
--- a/samples/sample_full_pbr.cpp
+++ b/samples/sample_full_pbr.cpp
@@ -37,7 +37,7 @@
 #include <math/mat4.h>
 #include <math/vec3.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <stb_image.h>
 
diff --git a/samples/sample_normal_map.cpp b/samples/sample_normal_map.cpp
index 684b398..f31aa24 100644
--- a/samples/sample_normal_map.cpp
+++ b/samples/sample_normal_map.cpp
@@ -19,7 +19,7 @@
 #include <map>
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/samples/suzanne.cpp b/samples/suzanne.cpp
index dae42d3..9e695c7 100644
--- a/samples/suzanne.cpp
+++ b/samples/suzanne.cpp
@@ -35,7 +35,7 @@
 #include <filamentapp/FilamentApp.h>
 #include <filamentapp/IBL.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/samples/viewtest.cpp b/samples/viewtest.cpp
index 26ee9ac..1467589 100644
--- a/samples/viewtest.cpp
+++ b/samples/viewtest.cpp
@@ -26,7 +26,7 @@
 
 #include <utils/EntityManager.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <iostream>
 #include <string>
diff --git a/tools/cmgen/src/cmgen.cpp b/tools/cmgen/src/cmgen.cpp
index 04eb151..c37dff6 100644
--- a/tools/cmgen/src/cmgen.cpp
+++ b/tools/cmgen/src/cmgen.cpp
@@ -44,7 +44,7 @@
 
 #include <string.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 
 using namespace filament::math;
diff --git a/tools/cso-lut/src/main.cpp b/tools/cso-lut/src/main.cpp
index b45a607..0233619 100644
--- a/tools/cso-lut/src/main.cpp
+++ b/tools/cso-lut/src/main.cpp
@@ -31,7 +31,7 @@
 #include <utils/compiler.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 using namespace filament::math;
 using namespace image;
diff --git a/tools/filamesh/src/main.cpp b/tools/filamesh/src/main.cpp
index 541d7ff..e9ba1b5 100644
--- a/tools/filamesh/src/main.cpp
+++ b/tools/filamesh/src/main.cpp
@@ -30,7 +30,7 @@
 
 #include <filameshio/filamesh.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 using namespace filamesh;
 using namespace filament::math;
diff --git a/tools/glslminifier/src/main.cpp b/tools/glslminifier/src/main.cpp
index a6d2f56..4f79833 100644
--- a/tools/glslminifier/src/main.cpp
+++ b/tools/glslminifier/src/main.cpp
@@ -18,7 +18,7 @@
 
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <fstream>
 #include <iostream>
diff --git a/tools/matc/src/matc/CommandlineConfig.cpp b/tools/matc/src/matc/CommandlineConfig.cpp
index 78460ad..85c4118 100644
--- a/tools/matc/src/matc/CommandlineConfig.cpp
+++ b/tools/matc/src/matc/CommandlineConfig.cpp
@@ -18,7 +18,7 @@
 
 #include <private/filament/Variant.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/tools/matc/src/matc/MaterialCompiler.cpp b/tools/matc/src/matc/MaterialCompiler.cpp
index 2785a73..3685dd8 100644
--- a/tools/matc/src/matc/MaterialCompiler.cpp
+++ b/tools/matc/src/matc/MaterialCompiler.cpp
@@ -33,7 +33,7 @@
 #include "JsonishParser.h"
 #include "ParametersProcessor.h"
 
-#include <GlslangToSpv.h>
+#include <glslang/SPIRV/GlslangToSpv.h>
 
 #include "sca/builtinResource.h"
 
diff --git a/tools/matinfo/src/main.cpp b/tools/matinfo/src/main.cpp
index 15a4583..368d7a6 100644
--- a/tools/matinfo/src/main.cpp
+++ b/tools/matinfo/src/main.cpp
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
@@ -25,7 +25,7 @@
 #include <matdbg/ShaderInfo.h>
 #include <matdbg/TextWriter.h>
 
-#include <spirv_glsl.hpp>
+#include <spirv_cross/spirv_glsl.hpp>
 #include <spirv-tools/libspirv.h>
 
 #include <chrono>
diff --git a/tools/mipgen/src/main.cpp b/tools/mipgen/src/main.cpp
index c7afabf..8a87dce 100644
--- a/tools/mipgen/src/main.cpp
+++ b/tools/mipgen/src/main.cpp
@@ -26,7 +26,7 @@
 
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <fstream>
 #include <iostream>
diff --git a/tools/normal-blending/src/main.cpp b/tools/normal-blending/src/main.cpp
index e3cadf4..76ec57b 100644
--- a/tools/normal-blending/src/main.cpp
+++ b/tools/normal-blending/src/main.cpp
@@ -25,7 +25,7 @@
 
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 using namespace filament::math;
 using namespace image;
diff --git a/tools/resgen/src/main.cpp b/tools/resgen/src/main.cpp
index 8cae171..26b11ad 100644
--- a/tools/resgen/src/main.cpp
+++ b/tools/resgen/src/main.cpp
@@ -16,7 +16,7 @@
 
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <algorithm>
 #include <fstream>
diff --git a/tools/rgb-to-lmsr/src/main.cpp b/tools/rgb-to-lmsr/src/main.cpp
index 8d4f3cd..07e8e6b 100644
--- a/tools/rgb-to-lmsr/src/main.cpp
+++ b/tools/rgb-to-lmsr/src/main.cpp
@@ -18,7 +18,7 @@
 #include <iomanip>
 #include <iostream>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/tools/roughness-prefilter/src/main.cpp b/tools/roughness-prefilter/src/main.cpp
index e6ed1d7..ed15d6d 100644
--- a/tools/roughness-prefilter/src/main.cpp
+++ b/tools/roughness-prefilter/src/main.cpp
@@ -33,7 +33,7 @@
 #include <utils/JobSystem.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 using namespace filament::math;
 using namespace image;
diff --git a/tools/specular-color/src/main.cpp b/tools/specular-color/src/main.cpp
index cfbf3c1..8676466 100644
--- a/tools/specular-color/src/main.cpp
+++ b/tools/specular-color/src/main.cpp
@@ -23,7 +23,7 @@
 
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 #include <utils/string.h>
diff --git a/tools/uberz/src/main.cpp b/tools/uberz/src/main.cpp
index 43e2005..365222b 100644
--- a/tools/uberz/src/main.cpp
+++ b/tools/uberz/src/main.cpp
@@ -16,7 +16,7 @@
 
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <fstream>
 #include <iostream>
diff --git a/filament/backend/test/ShaderGenerator.cpp b/filament/backend/test/ShaderGenerator.cpp
index df67bee..7797c6b 100644
--- a/filament/backend/test/ShaderGenerator.cpp
+++ b/filament/backend/test/ShaderGenerator.cpp
@@ -17,10 +17,10 @@
 #include "ShaderGenerator.h"
 
 #include <glslang/SPIRV/GlslangToSpv.h>
-#include <SPVRemapper.h>
+#include <glslang/SPIRV/SPVRemapper.h>
 
 #include <spirv_cross/spirv_glsl.hpp>
-#include <spirv_msl.hpp>
+#include <spirv_cross/spirv_msl.hpp>
 
 #include "../src/GLSLPostProcessor.h"
 
diff --git a/libs/filamentapp/src/FilamentApp.cpp b/libs/filamentapp/src/FilamentApp.cpp
index 2e7bae9..125d565 100644
--- a/libs/filamentapp/src/FilamentApp.cpp
+++ b/libs/filamentapp/src/FilamentApp.cpp
@@ -51,6 +51,7 @@
 #include <filamentapp/Cube.h>
 #include <filamentapp/NativeWindowHelper.h>
 
+#define STB_IMAGE_IMPLEMENTATION
 #include <stb_image.h>
 
 #include "generated/resources/filamentapp.h"
diff --git a/libs/filamentapp/src/MeshAssimp.cpp b/libs/filamentapp/src/MeshAssimp.cpp
index 5e3f403..2bf74af 100644
--- a/libs/filamentapp/src/MeshAssimp.cpp
+++ b/libs/filamentapp/src/MeshAssimp.cpp
@@ -51,7 +51,12 @@
 #include <assimp/postprocess.h>
 #include <assimp/cimport.h>
 #include <assimp/scene.h>
+#include <assimp/material.h>
+#include <assimp/GltfMaterial.h>
 #include <assimp/pbrmaterial.h>
+#if !defined(_AI_MATKEY_GLTF_TEXTURE_TEXCOORD_BASE)
+#define _AI_MATKEY_GLTF_TEXTURE_TEXCOORD_BASE "$tex.file.texCoord"
+#endif
 
 #include <stb_image.h>
 
