diff --git a/filament/backend/test/Arguments.cpp b/filament/backend/test/Arguments.cpp
index 49fa4b7..1332d78 100644
--- a/filament/backend/test/Arguments.cpp
+++ b/filament/backend/test/Arguments.cpp
@@ -16,7 +16,7 @@
 
 #include "PlatformRunner.h"
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <iostream>
 #include <string>
diff --git a/libs/filamat/src/GLSLPostProcessor.cpp b/libs/filamat/src/GLSLPostProcessor.cpp
index 3a4a482..4210a36 100644
--- a/libs/filamat/src/GLSLPostProcessor.cpp
+++ b/libs/filamat/src/GLSLPostProcessor.cpp
@@ -16,15 +16,15 @@
 
 #include "GLSLPostProcessor.h"
 
-#include <GlslangToSpv.h>
-#include <SPVRemapper.h>
+#include <glslang/SPIRV/GlslangToSpv.h>
+#include <glslang/SPIRV/SPVRemapper.h>
 #include <spirv-tools/libspirv.hpp>
 
 #include <spirv_glsl.hpp>
 #include <spirv_msl.hpp>
 
 #include "backend/DriverEnums.h"
-#include "sca/builtinResource.h"
+#include <glslang/Public/ResourceLimits.h>
 #include "sca/GLSLTools.h"
 
 #include "shaders/CodeGenerator.h"
@@ -363,7 +363,7 @@ bool GLSLPostProcessor::process(const std::string& inputShader, Config const& co
         msg = EShMessages(Type(msg) | Type(EShMessages::EShMsgVulkanRules));
     }
 
-    bool const ok = tShader.parse(&DefaultTBuiltInResource, internalConfig.langVersion, false, msg);
+    bool const ok = tShader.parse(GetDefaultResources(), internalConfig.langVersion, false, msg);
     if (!ok) {
         slog.e << tShader.getInfoLog() << io::endl;
         return false;
@@ -446,7 +446,7 @@ void GLSLPostProcessor::preprocessOptimization(glslang::TShader& tShader,
     const int version = GLSLTools::getGlslDefaultVersion(config.shaderModel);
     EShMessages const msg =
             GLSLTools::glslangFlagsFromTargetApi(config.targetApi, config.targetLanguage);
-    bool ok = tShader.preprocess(&DefaultTBuiltInResource, version, ENoProfile, false, false,
+    bool ok = tShader.preprocess(GetDefaultResources(), version, ENoProfile, false, false,
             msg, &glsl, forbidIncluder);
 
     if (!ok) {
@@ -465,7 +465,7 @@ void GLSLPostProcessor::preprocessOptimization(glslang::TShader& tShader,
         spirvShader.setStrings(&shaderCString, 1);
         GLSLTools::prepareShaderParser(config.targetApi, config.targetLanguage, spirvShader,
                 internalConfig.shLang, internalConfig.langVersion);
-        ok = spirvShader.parse(&DefaultTBuiltInResource, internalConfig.langVersion, false, msg);
+        ok = spirvShader.parse(GetDefaultResources(), internalConfig.langVersion, false, msg);
         program.addShader(&spirvShader);
         // Even though we only have a single shader stage, linking is still necessary to finalize
         // SPIR-V types
diff --git a/libs/filamat/src/GLSLPostProcessor.h b/libs/filamat/src/GLSLPostProcessor.h
index c13dece..c2fb87c 100644
--- a/libs/filamat/src/GLSLPostProcessor.h
+++ b/libs/filamat/src/GLSLPostProcessor.h
@@ -25,7 +25,7 @@
 
 #include <spirv-tools/optimizer.hpp>
 
-#include <ShaderLang.h>
+#include <glslang/Public/ShaderLang.h>
 
 #include <backend/DriverEnums.h>
 
diff --git a/libs/filamat/src/sca/ASTHelpers.cpp b/libs/filamat/src/sca/ASTHelpers.cpp
index d82831a..19df976 100644
--- a/libs/filamat/src/sca/ASTHelpers.cpp
+++ b/libs/filamat/src/sca/ASTHelpers.cpp
@@ -18,8 +18,8 @@
 
 #include "GLSLTools.h"
 
-#include <intermediate.h>
-#include <localintermediate.h>
+#include <glslang/Include/intermediate.h>
+#include <glslang/MachineIndependent/localintermediate.h>
 
 #include <utils/Log.h>
 
diff --git a/libs/filamat/src/sca/ASTHelpers.h b/libs/filamat/src/sca/ASTHelpers.h
index 19ec1ce..680466f 100644
--- a/libs/filamat/src/sca/ASTHelpers.h
+++ b/libs/filamat/src/sca/ASTHelpers.h
@@ -19,7 +19,7 @@
 
 #include <string>
 #include <vector>
-#include <intermediate.h>
+#include <glslang/Include/intermediate.h>
 
 class TIntermNode;
 
diff --git a/libs/filamat/src/sca/GLSLTools.cpp b/libs/filamat/src/sca/GLSLTools.cpp
index eeecf89..1b27999 100644
--- a/libs/filamat/src/sca/GLSLTools.cpp
+++ b/libs/filamat/src/sca/GLSLTools.cpp
@@ -25,10 +25,10 @@
 #include "ASTHelpers.h"
 
 // GLSLANG headers
-#include <InfoSink.h>
-#include <localintermediate.h>
+#include <glslang/Include/InfoSink.h>
+#include <glslang/MachineIndependent/localintermediate.h>
 
-#include "builtinResource.h"
+#include <glslang/Public/ResourceLimits.h>
 
 using namespace utils;
 using namespace glslang;
@@ -173,7 +173,7 @@ bool GLSLTools::analyzeComputeShader(const std::string& shaderCode,
     GLSLangCleaner const cleaner;
     const int version = getGlslDefaultVersion(model);
     EShMessages const msg = glslangFlagsFromTargetApi(targetApi, targetLanguage);
-    bool const ok = tShader.parse(&DefaultTBuiltInResource, version, false, msg);
+    bool const ok = tShader.parse(GetDefaultResources(), version, false, msg);
     if (!ok) {
         utils::slog.e << "ERROR: Unable to parse compute shader:" << utils::io::endl;
         utils::slog.e << tShader.getInfoLog() << utils::io::endl;
@@ -211,7 +211,7 @@ std::optional<GLSLTools::FragmentShaderInfo> GLSLTools::analyzeFragmentShader(
     GLSLangCleaner const cleaner;
     const int version = getGlslDefaultVersion(model);
     EShMessages const msg = glslangFlagsFromTargetApi(targetApi, targetLanguage);
-    bool const ok = tShader.parse(&DefaultTBuiltInResource, version, false, msg);
+    bool const ok = tShader.parse(GetDefaultResources(), version, false, msg);
     if (!ok) {
         utils::slog.e << "ERROR: Unable to parse fragment shader:" << utils::io::endl;
         utils::slog.e << tShader.getInfoLog() << utils::io::endl;
@@ -291,7 +291,7 @@ bool GLSLTools::analyzeVertexShader(const std::string& shaderCode,
     GLSLangCleaner const cleaner;
     const int version = getGlslDefaultVersion(model);
     EShMessages const msg = glslangFlagsFromTargetApi(targetApi, targetLanguage);
-    bool const ok = tShader.parse(&DefaultTBuiltInResource, version, false, msg);
+    bool const ok = tShader.parse(GetDefaultResources(), version, false, msg);
     if (!ok) {
         utils::slog.e << "ERROR: Unable to parse vertex shader" << utils::io::endl;
         utils::slog.e << tShader.getInfoLog() << utils::io::endl;
@@ -342,7 +342,7 @@ bool GLSLTools::findProperties(
     GLSLangCleaner const cleaner;
     const int version = getGlslDefaultVersion(model);
     EShMessages const msg = glslangFlagsFromTargetApi(targetApi, targetLanguage);
-    const TBuiltInResource* builtins = &DefaultTBuiltInResource;
+    const TBuiltInResource* builtins = GetDefaultResources();
     bool const ok = tShader.parse(builtins, version, false, msg);
     if (!ok) {
         // Even with all properties set the shader doesn't build. This is likely a syntax error
diff --git a/libs/filamat/src/sca/GLSLTools.h b/libs/filamat/src/sca/GLSLTools.h
index 1c7cf16..fb2a610 100644
--- a/libs/filamat/src/sca/GLSLTools.h
+++ b/libs/filamat/src/sca/GLSLTools.h
@@ -25,7 +25,7 @@
 
 #include <filamat/MaterialBuilder.h>
 
-#include <ShaderLang.h>
+#include <glslang/Public/ShaderLang.h>
 
 class TIntermNode;
 
diff --git a/libs/geometry/src/MikktspaceImpl.cpp b/libs/geometry/src/MikktspaceImpl.cpp
index 5c5a1b5..4c6dd9b 100644
--- a/libs/geometry/src/MikktspaceImpl.cpp
+++ b/libs/geometry/src/MikktspaceImpl.cpp
@@ -21,7 +21,7 @@
 #include <utils/Panic.h>
 
 #include <meshoptimizer.h>
-#include <mikktspace/mikktspace.h>
+#include <mikktspace.h>
 
 #include <vector>
 
diff --git a/libs/imageio/src/BasisEncoder.cpp b/libs/imageio/src/BasisEncoder.cpp
index 0c24e71..fb4eea1 100644
--- a/libs/imageio/src/BasisEncoder.cpp
+++ b/libs/imageio/src/BasisEncoder.cpp
@@ -22,7 +22,7 @@
 
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Warray-bounds"
-#include <basisu_comp.h>
+#include <basisu/encoder/basisu_comp.h>
 #pragma clang diagnostic pop
 
 namespace image {
diff --git a/libs/ktxreader/src/Ktx2Reader.cpp b/libs/ktxreader/src/Ktx2Reader.cpp
index 5d53861..f72c108 100644
--- a/libs/ktxreader/src/Ktx2Reader.cpp
+++ b/libs/ktxreader/src/Ktx2Reader.cpp
@@ -26,7 +26,7 @@
 
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Warray-bounds"
-#include <basisu_transcoder.h>
+#include <basisu/transcoder/basisu_transcoder.h>
 #pragma clang diagnostic pop
 
 using namespace basist;
diff --git a/libs/matdbg/src/ShaderReplacer.cpp b/libs/matdbg/src/ShaderReplacer.cpp
index 9cd8ba4..d2b4d0f 100644
--- a/libs/matdbg/src/ShaderReplacer.cpp
+++ b/libs/matdbg/src/ShaderReplacer.cpp
@@ -27,9 +27,9 @@
 
 #include <sstream>
 
-#include <GlslangToSpv.h>
+#include <glslang/SPIRV/GlslangToSpv.h>
 
-#include "sca/builtinResource.h"
+#include <glslang/Public/ResourceLimits.h>
 #include "sca/GLSLTools.h"
 
 #include "eiff/ChunkContainer.h"
@@ -225,7 +225,7 @@ bool ShaderReplacer::replaceSpirv(ShaderModel shaderModel, Variant variant,
         GLSLTools::prepareShaderParser(targetApi,
                 MaterialBuilder::TargetLanguage::SPIRV, tShader, shLang, version);
 
-        const bool ok = tShader.parse(&DefaultTBuiltInResource, version, false, msg);
+        const bool ok = tShader.parse(GetDefaultResources(), version, false, msg);
         if (!ok) {
             slog.e << "ShaderReplacer parse:\n" << tShader.getInfoLog() << io::endl;
             return false;
diff --git a/libs/viewer/src/RemoteServer.cpp b/libs/viewer/src/RemoteServer.cpp
index 441c730..89c492f 100644
--- a/libs/viewer/src/RemoteServer.cpp
+++ b/libs/viewer/src/RemoteServer.cpp
@@ -208,8 +208,8 @@ bool MessageReceiver::handleData(CivetServer* server, struct mg_connection* conn
 
 void MessageSender::sendMessage(const char* label, const char* buffer, size_t bufsize) {
     for (auto iter : connections) {
-        mg_websocket_write(iter.first, 0x80, label, strlen(label) + 1);
-        mg_websocket_write(iter.first, 0x80, buffer, bufsize);
+        mg_websocket_write(const_cast<mg_connection*>(iter.first), 0x80, label, strlen(label) + 1);
+        mg_websocket_write(const_cast<mg_connection*>(iter.first), 0x80, buffer, bufsize);
     }
 }
 
diff --git a/samples/frame_generator.cpp b/samples/frame_generator.cpp
index a6a2164..3fb7e80 100644
--- a/samples/frame_generator.cpp
+++ b/samples/frame_generator.cpp
@@ -25,7 +25,7 @@
 #include <utility>
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/samples/gltf_instances.cpp b/samples/gltf_instances.cpp
index 688272d..5210ebd 100644
--- a/samples/gltf_instances.cpp
+++ b/samples/gltf_instances.cpp
@@ -33,7 +33,7 @@
 
 #include <camutils/Manipulator.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/NameComponentManager.h>
 
diff --git a/samples/gltf_viewer.cpp b/samples/gltf_viewer.cpp
index 77aee47..b272d0d 100644
--- a/samples/gltf_viewer.cpp
+++ b/samples/gltf_viewer.cpp
@@ -43,7 +43,7 @@
 
 #include <private/filament/EngineEnums.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/NameComponentManager.h>
 #include <utils/Log.h>
diff --git a/samples/heightfield.cpp b/samples/heightfield.cpp
index 855dabf..92c98ad 100644
--- a/samples/heightfield.cpp
+++ b/samples/heightfield.cpp
@@ -32,7 +32,7 @@
 #include <utils/JobSystem.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <filamentapp/FilamentApp.h>
 
diff --git a/samples/hellomorphing.cpp b/samples/hellomorphing.cpp
index 54ce244..eb10396 100644
--- a/samples/hellomorphing.cpp
+++ b/samples/hellomorphing.cpp
@@ -29,7 +29,7 @@
 #include <utils/EntityManager.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
diff --git a/samples/hellopbr.cpp b/samples/hellopbr.cpp
index f5cae37..08c63d8 100644
--- a/samples/hellopbr.cpp
+++ b/samples/hellopbr.cpp
@@ -29,7 +29,7 @@
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <iostream>
 
diff --git a/samples/helloskinning.cpp b/samples/helloskinning.cpp
index f3d5046..07ee5ab 100644
--- a/samples/helloskinning.cpp
+++ b/samples/helloskinning.cpp
@@ -35,7 +35,7 @@
 #include <cmath>
 #include <iostream>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include "generated/resources/resources.h"
 
diff --git a/samples/helloskinningbuffer.cpp b/samples/helloskinningbuffer.cpp
index 175c62c..59419e3 100644
--- a/samples/helloskinningbuffer.cpp
+++ b/samples/helloskinningbuffer.cpp
@@ -29,7 +29,7 @@
 #include <utils/EntityManager.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
diff --git a/samples/helloskinningbuffer_morebones.cpp b/samples/helloskinningbuffer_morebones.cpp
index 184d799..2f7051c 100644
--- a/samples/helloskinningbuffer_morebones.cpp
+++ b/samples/helloskinningbuffer_morebones.cpp
@@ -29,7 +29,7 @@
 #include <utils/EntityManager.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
diff --git a/samples/hellostereo.cpp b/samples/hellostereo.cpp
index fff1616..ef377d7 100644
--- a/samples/hellostereo.cpp
+++ b/samples/hellostereo.cpp
@@ -37,7 +37,7 @@
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <iostream>
 #include <vector>
diff --git a/samples/hellotriangle.cpp b/samples/hellotriangle.cpp
index 4477429..67959a5 100644
--- a/samples/hellotriangle.cpp
+++ b/samples/hellotriangle.cpp
@@ -31,7 +31,7 @@
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <cmath>
 #include <iostream>
diff --git a/samples/image_viewer.cpp b/samples/image_viewer.cpp
index d380aae..c3266fc 100644
--- a/samples/image_viewer.cpp
+++ b/samples/image_viewer.cpp
@@ -36,7 +36,7 @@
 
 #include <camutils/Manipulator.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <math/half.h>
 #include <math/vec3.h>
diff --git a/samples/lightbulb.cpp b/samples/lightbulb.cpp
index 5c5d735..385a5ac 100644
--- a/samples/lightbulb.cpp
+++ b/samples/lightbulb.cpp
@@ -19,7 +19,7 @@
 #include <map>
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/EntityManager.h>
 #include <utils/Path.h>
diff --git a/samples/material_sandbox.cpp b/samples/material_sandbox.cpp
index 7778246..2c8e881 100644
--- a/samples/material_sandbox.cpp
+++ b/samples/material_sandbox.cpp
@@ -19,7 +19,7 @@
 #include <map>
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <imgui.h>
 
diff --git a/samples/rendertarget.cpp b/samples/rendertarget.cpp
index 9ff53cc..8cf3de9 100644
--- a/samples/rendertarget.cpp
+++ b/samples/rendertarget.cpp
@@ -35,7 +35,7 @@
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <iostream>
 
diff --git a/samples/sample_cloth.cpp b/samples/sample_cloth.cpp
index b7a5627..ec32060 100644
--- a/samples/sample_cloth.cpp
+++ b/samples/sample_cloth.cpp
@@ -19,7 +19,7 @@
 #include <map>
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/samples/sample_full_pbr.cpp b/samples/sample_full_pbr.cpp
index 77c08ec..edc284f 100644
--- a/samples/sample_full_pbr.cpp
+++ b/samples/sample_full_pbr.cpp
@@ -37,7 +37,7 @@
 #include <math/mat4.h>
 #include <math/vec3.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <stb_image.h>
 
diff --git a/samples/sample_normal_map.cpp b/samples/sample_normal_map.cpp
index 684b398..f31aa24 100644
--- a/samples/sample_normal_map.cpp
+++ b/samples/sample_normal_map.cpp
@@ -19,7 +19,7 @@
 #include <map>
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/samples/skinningtest.cpp b/samples/skinningtest.cpp
index a2da0b1..3c173d2 100644
--- a/samples/skinningtest.cpp
+++ b/samples/skinningtest.cpp
@@ -30,7 +30,7 @@
 #include <utils/EntityManager.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
diff --git a/samples/suzanne.cpp b/samples/suzanne.cpp
index dae42d3..9e695c7 100644
--- a/samples/suzanne.cpp
+++ b/samples/suzanne.cpp
@@ -35,7 +35,7 @@
 #include <filamentapp/FilamentApp.h>
 #include <filamentapp/IBL.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/samples/viewtest.cpp b/samples/viewtest.cpp
index 26ee9ac..1467589 100644
--- a/samples/viewtest.cpp
+++ b/samples/viewtest.cpp
@@ -26,7 +26,7 @@
 
 #include <utils/EntityManager.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <iostream>
 #include <string>
diff --git a/tools/cmgen/src/cmgen.cpp b/tools/cmgen/src/cmgen.cpp
index 04eb151..c37dff6 100644
--- a/tools/cmgen/src/cmgen.cpp
+++ b/tools/cmgen/src/cmgen.cpp
@@ -44,7 +44,7 @@
 
 #include <string.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 
 using namespace filament::math;
diff --git a/tools/cso-lut/src/main.cpp b/tools/cso-lut/src/main.cpp
index b45a607..0233619 100644
--- a/tools/cso-lut/src/main.cpp
+++ b/tools/cso-lut/src/main.cpp
@@ -31,7 +31,7 @@
 #include <utils/compiler.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 using namespace filament::math;
 using namespace image;
diff --git a/tools/filamesh/src/main.cpp b/tools/filamesh/src/main.cpp
index 541d7ff..e9ba1b5 100644
--- a/tools/filamesh/src/main.cpp
+++ b/tools/filamesh/src/main.cpp
@@ -30,7 +30,7 @@
 
 #include <filameshio/filamesh.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 using namespace filamesh;
 using namespace filament::math;
diff --git a/tools/glslminifier/src/main.cpp b/tools/glslminifier/src/main.cpp
index a6d2f56..4f79833 100644
--- a/tools/glslminifier/src/main.cpp
+++ b/tools/glslminifier/src/main.cpp
@@ -18,7 +18,7 @@
 
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <fstream>
 #include <iostream>
diff --git a/tools/matc/src/matc/CommandlineConfig.cpp b/tools/matc/src/matc/CommandlineConfig.cpp
index 6ca6315..11b90b9 100644
--- a/tools/matc/src/matc/CommandlineConfig.cpp
+++ b/tools/matc/src/matc/CommandlineConfig.cpp
@@ -18,7 +18,7 @@
 
 #include <private/filament/Variant.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/tools/matc/src/matc/MaterialCompiler.cpp b/tools/matc/src/matc/MaterialCompiler.cpp
index 93d0fed..eca6122 100644
--- a/tools/matc/src/matc/MaterialCompiler.cpp
+++ b/tools/matc/src/matc/MaterialCompiler.cpp
@@ -34,10 +34,9 @@
 #include "JsonishParser.h"
 #include "ParametersProcessor.h"
 
-#include <GlslangToSpv.h>
-#include "glslang/Include/intermediate.h"
-
-#include "sca/builtinResource.h"
+#include <glslang/SPIRV/GlslangToSpv.h>
+#include <glslang/Public/ShaderLang.h>
+#include <glslang/Public/ResourceLimits.h>
 
 #include <smolv.h>
 
@@ -583,7 +582,7 @@ bool MaterialCompiler::compileRawShader(const char* glsl, size_t size, bool isDe
 
     tShader.setAutoMapBindings(true);
 
-    bool const parseOk = tShader.parse(&DefaultTBuiltInResource, version, false, msg);
+    bool const parseOk = tShader.parse(GetDefaultResources(), version, false, msg);
     if (!parseOk) {
         std::cerr << "ERROR: Unable to parse " << ext << ":" << std::endl;
         std::cerr << tShader.getInfoLog() << std::endl;
diff --git a/tools/matedit/src/main.cpp b/tools/matedit/src/main.cpp
index 50b4791..3632102 100644
--- a/tools/matedit/src/main.cpp
+++ b/tools/matedit/src/main.cpp
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include "ExternalCompile.h"
 
diff --git a/tools/matinfo/src/main.cpp b/tools/matinfo/src/main.cpp
index ea2d1a0..d60f63d 100644
--- a/tools/matinfo/src/main.cpp
+++ b/tools/matinfo/src/main.cpp
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/tools/mipgen/src/main.cpp b/tools/mipgen/src/main.cpp
index c7afabf..8a87dce 100644
--- a/tools/mipgen/src/main.cpp
+++ b/tools/mipgen/src/main.cpp
@@ -26,7 +26,7 @@
 
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <fstream>
 #include <iostream>
diff --git a/tools/normal-blending/src/main.cpp b/tools/normal-blending/src/main.cpp
index e3cadf4..76ec57b 100644
--- a/tools/normal-blending/src/main.cpp
+++ b/tools/normal-blending/src/main.cpp
@@ -25,7 +25,7 @@
 
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 using namespace filament::math;
 using namespace image;
diff --git a/tools/resgen/src/main.cpp b/tools/resgen/src/main.cpp
index 1f475c4..d97aaeb 100644
--- a/tools/resgen/src/main.cpp
+++ b/tools/resgen/src/main.cpp
@@ -16,7 +16,7 @@
 
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <algorithm>
 #include <cstdint>
diff --git a/tools/rgb-to-lmsr/src/main.cpp b/tools/rgb-to-lmsr/src/main.cpp
index 8d4f3cd..07e8e6b 100644
--- a/tools/rgb-to-lmsr/src/main.cpp
+++ b/tools/rgb-to-lmsr/src/main.cpp
@@ -18,7 +18,7 @@
 #include <iomanip>
 #include <iostream>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/tools/roughness-prefilter/src/main.cpp b/tools/roughness-prefilter/src/main.cpp
index e6ed1d7..ed15d6d 100644
--- a/tools/roughness-prefilter/src/main.cpp
+++ b/tools/roughness-prefilter/src/main.cpp
@@ -33,7 +33,7 @@
 #include <utils/JobSystem.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 using namespace filament::math;
 using namespace image;
diff --git a/tools/specular-color/src/main.cpp b/tools/specular-color/src/main.cpp
index cfbf3c1..8676466 100644
--- a/tools/specular-color/src/main.cpp
+++ b/tools/specular-color/src/main.cpp
@@ -23,7 +23,7 @@
 
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 #include <utils/string.h>
diff --git a/tools/uberz/src/main.cpp b/tools/uberz/src/main.cpp
index 43e2005..365222b 100644
--- a/tools/uberz/src/main.cpp
+++ b/tools/uberz/src/main.cpp
@@ -16,7 +16,7 @@
 
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <fstream>
 #include <iostream>
