diff --git a/filament/backend/test/Arguments.cpp b/filament/backend/test/Arguments.cpp
index 49fa4b7..1332d78 100644
--- a/filament/backend/test/Arguments.cpp
+++ b/filament/backend/test/Arguments.cpp
@@ -16,7 +16,7 @@
 
 #include "PlatformRunner.h"
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <iostream>
 #include <string>
diff --git a/filament/backend/test/ShaderGenerator.cpp b/filament/backend/test/ShaderGenerator.cpp
index 7d0ad71..64504f1 100644
--- a/filament/backend/test/ShaderGenerator.cpp
+++ b/filament/backend/test/ShaderGenerator.cpp
@@ -24,7 +24,7 @@
 
 #include "../src/GLSLPostProcessor.h"
 
-#include "builtinResource.h"
+#include <glslang/Include/ResourceLimits.h>
 
 #include <utils/FixedCapacityVector.h>
 
diff --git a/libs/filamat/src/GLSLPostProcessor.cpp b/libs/filamat/src/GLSLPostProcessor.cpp
index 3a4a482..653eb73 100644
--- a/libs/filamat/src/GLSLPostProcessor.cpp
+++ b/libs/filamat/src/GLSLPostProcessor.cpp
@@ -16,15 +16,15 @@
 
 #include "GLSLPostProcessor.h"
 
-#include <GlslangToSpv.h>
-#include <SPVRemapper.h>
+#include <glslang/SPIRV/GlslangToSpv.h>
+#include <glslang/SPIRV/SPVRemapper.h>
 #include <spirv-tools/libspirv.hpp>
 
 #include <spirv_glsl.hpp>
 #include <spirv_msl.hpp>
 
 #include "backend/DriverEnums.h"
-#include "sca/builtinResource.h"
+#include <glslang/Include/ResourceLimits.h>
 #include "sca/GLSLTools.h"
 
 #include "shaders/CodeGenerator.h"
diff --git a/libs/filamat/src/GLSLPostProcessor.h b/libs/filamat/src/GLSLPostProcessor.h
index c13dece..c2fb87c 100644
--- a/libs/filamat/src/GLSLPostProcessor.h
+++ b/libs/filamat/src/GLSLPostProcessor.h
@@ -25,7 +25,7 @@
 
 #include <spirv-tools/optimizer.hpp>
 
-#include <ShaderLang.h>
+#include <glslang/Public/ShaderLang.h>
 
 #include <backend/DriverEnums.h>
 
diff --git a/libs/filamat/src/sca/ASTHelpers.cpp b/libs/filamat/src/sca/ASTHelpers.cpp
index d82831a..19df976 100644
--- a/libs/filamat/src/sca/ASTHelpers.cpp
+++ b/libs/filamat/src/sca/ASTHelpers.cpp
@@ -18,8 +18,8 @@
 
 #include "GLSLTools.h"
 
-#include <intermediate.h>
-#include <localintermediate.h>
+#include <glslang/Include/intermediate.h>
+#include <glslang/MachineIndependent/localintermediate.h>
 
 #include <utils/Log.h>
 
diff --git a/libs/filamat/src/sca/ASTHelpers.h b/libs/filamat/src/sca/ASTHelpers.h
index 19ec1ce..680466f 100644
--- a/libs/filamat/src/sca/ASTHelpers.h
+++ b/libs/filamat/src/sca/ASTHelpers.h
@@ -19,7 +19,7 @@
 
 #include <string>
 #include <vector>
-#include <intermediate.h>
+#include <glslang/Include/intermediate.h>
 
 class TIntermNode;
 
diff --git a/libs/filamat/src/sca/GLSLTools.cpp b/libs/filamat/src/sca/GLSLTools.cpp
index eeecf89..19aee05 100644
--- a/libs/filamat/src/sca/GLSLTools.cpp
+++ b/libs/filamat/src/sca/GLSLTools.cpp
@@ -25,10 +25,10 @@
 #include "ASTHelpers.h"
 
 // GLSLANG headers
-#include <InfoSink.h>
-#include <localintermediate.h>
+#include <glslang/Include/InfoSink.h>
+#include <glslang/MachineIndependent/localintermediate.h>
 
-#include "builtinResource.h"
+#include <glslang/Include/ResourceLimits.h>
 
 using namespace utils;
 using namespace glslang;
diff --git a/libs/filamat/src/sca/GLSLTools.h b/libs/filamat/src/sca/GLSLTools.h
index 1c7cf16..fb2a610 100644
--- a/libs/filamat/src/sca/GLSLTools.h
+++ b/libs/filamat/src/sca/GLSLTools.h
@@ -25,7 +25,7 @@
 
 #include <filamat/MaterialBuilder.h>
 
-#include <ShaderLang.h>
+#include <glslang/Public/ShaderLang.h>
 
 class TIntermNode;
 
diff --git a/libs/filamat/src/sca/builtinResource.h b/libs/filamat/src/sca/builtinResource.h
deleted file mode 100644
index d58f2a9..0000000
--- a/libs/filamat/src/sca/builtinResource.h
+++ /dev/null
@@ -1,134 +0,0 @@
-/*
- * Copyright (C) 2018 The Android Open Source Project
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-#include "glslang/Include/intermediate.h"
-
-const TBuiltInResource DefaultTBuiltInResource = {
-        /* .MaxLights = */ 32,
-        /* .MaxClipPlanes = */ 6,
-        /* .MaxTextureUnits = */ 32,
-        /* .MaxTextureCoords = */ 32,
-        /* .MaxVertexAttribs = */ 64,
-        /* .MaxVertexUniformComponents = */ 4096,
-        /* .MaxVaryingFloats = */ 64,
-        /* .MaxVertexTextureImageUnits = */ 32,
-        /* .MaxCombinedTextureImageUnits = */ 80,
-        /* .MaxTextureImageUnits = */ 32,
-        /* .MaxFragmentUniformComponents = */ 4096,
-        /* .MaxDrawBuffers = */ 32,
-        /* .MaxVertexUniformVectors = */ 128,
-        /* .MaxVaryingVectors = */ 8,
-        /* .MaxFragmentUniformVectors = */ 16,
-        /* .MaxVertexOutputVectors = */ 16,
-        /* .MaxFragmentInputVectors = */ 15,
-        /* .MinProgramTexelOffset = */ -8,
-        /* .MaxProgramTexelOffset = */ 7,
-        /* .MaxClipDistances = */ 8,
-        /* .MaxComputeWorkGroupCountX = */ 65535,
-        /* .MaxComputeWorkGroupCountY = */ 65535,
-        /* .MaxComputeWorkGroupCountZ = */ 65535,
-        /* .MaxComputeWorkGroupSizeX = */ 1024,
-        /* .MaxComputeWorkGroupSizeY = */ 1024,
-        /* .MaxComputeWorkGroupSizeZ = */ 64,
-        /* .MaxComputeUniformComponents = */ 1024,
-        /* .MaxComputeTextureImageUnits = */ 16,
-        /* .MaxComputeImageUniforms = */ 8,
-        /* .MaxComputeAtomicCounters = */ 8,
-        /* .MaxComputeAtomicCounterBuffers = */ 1,
-        /* .MaxVaryingComponents = */ 60,
-        /* .MaxVertexOutputComponents = */ 64,
-        /* .MaxGeometryInputComponents = */ 64,
-        /* .MaxGeometryOutputComponents = */ 128,
-        /* .MaxFragmentInputComponents = */ 128,
-        /* .MaxImageUnits = */ 8,
-        /* .MaxCombinedImageUnitsAndFragmentOutputs = */ 8,
-        /* .MaxCombinedShaderOutputResources = */ 8,
-        /* .MaxImageSamples = */ 0,
-        /* .MaxVertexImageUniforms = */ 0,
-        /* .MaxTessControlImageUniforms = */ 0,
-        /* .MaxTessEvaluationImageUniforms = */ 0,
-        /* .MaxGeometryImageUniforms = */ 0,
-        /* .MaxFragmentImageUniforms = */ 8,
-        /* .MaxCombinedImageUniforms = */ 8,
-        /* .MaxGeometryTextureImageUnits = */ 16,
-        /* .MaxGeometryOutputVertices = */ 256,
-        /* .MaxGeometryTotalOutputComponents = */ 1024,
-        /* .MaxGeometryUniformComponents = */ 1024,
-        /* .MaxGeometryVaryingComponents = */ 64,
-        /* .MaxTessControlInputComponents = */ 128,
-        /* .MaxTessControlOutputComponents = */ 128,
-        /* .MaxTessControlTextureImageUnits = */ 16,
-        /* .MaxTessControlUniformComponents = */ 1024,
-        /* .MaxTessControlTotalOutputComponents = */ 4096,
-        /* .MaxTessEvaluationInputComponents = */ 128,
-        /* .MaxTessEvaluationOutputComponents = */ 128,
-        /* .MaxTessEvaluationTextureImageUnits = */ 16,
-        /* .MaxTessEvaluationUniformComponents = */ 1024,
-        /* .MaxTessPatchComponents = */ 120,
-        /* .MaxPatchVertices = */ 32,
-        /* .MaxTessGenLevel = */ 64,
-        /* .MaxViewports = */ 16,
-        /* .MaxVertexAtomicCounters = */ 0,
-        /* .MaxTessControlAtomicCounters = */ 0,
-        /* .MaxTessEvaluationAtomicCounters = */ 0,
-        /* .MaxGeometryAtomicCounters = */ 0,
-        /* .MaxFragmentAtomicCounters = */ 8,
-        /* .MaxCombinedAtomicCounters = */ 8,
-        /* .MaxAtomicCounterBindings = */ 1,
-        /* .MaxVertexAtomicCounterBuffers = */ 0,
-        /* .MaxTessControlAtomicCounterBuffers = */ 0,
-        /* .MaxTessEvaluationAtomicCounterBuffers = */ 0,
-        /* .MaxGeometryAtomicCounterBuffers = */ 0,
-        /* .MaxFragmentAtomicCounterBuffers = */ 1,
-        /* .MaxCombinedAtomicCounterBuffers = */ 1,
-        /* .MaxAtomicCounterBufferSize = */ 16384,
-        /* .MaxTransformFeedbackBuffers = */ 4,
-        /* .MaxTransformFeedbackInterleavedComponents = */ 64,
-        /* .MaxCullDistances = */ 8,
-        /* .MaxCombinedClipAndCullDistances = */ 8,
-        /* .MaxSamples = */ 4,
-        /* .maxMeshOutputVerticesNV = */ 256,
-        /* .maxMeshOutputPrimitivesNV = */ 512,
-        /* .maxMeshWorkGroupSizeX_NV = */ 32,
-        /* .maxMeshWorkGroupSizeY_NV = */ 1,
-        /* .maxMeshWorkGroupSizeZ_NV = */ 1,
-        /* .maxTaskWorkGroupSizeX_NV = */ 32,
-        /* .maxTaskWorkGroupSizeY_NV = */ 1,
-        /* .maxTaskWorkGroupSizeZ_NV = */ 1,
-        /* .maxMeshViewCountNV = */ 4,
-        /* .max_mesh_output_vertices_ext = */ 256,
-        /* .max_mesh_output_primitives_ext = */ 256,
-        /* .max_mesh_work_group_size_x_ext = */ 128,
-        /* .max_mesh_work_group_size_y_ext = */ 128,
-        /* .max_mesh_work_group_size_z_ext = */ 128,
-        /* .max_task_work_group_size_x_ext = */ 128,
-        /* .max_task_work_group_size_y_ext = */ 128,
-        /* .max_task_work_group_size_z_ext = */ 128,
-        /* .max_mesh_view_count_ext = */ 4,
-        /* .maxDualSourceDrawBuffersEXT = */ 1,
-
-        /* .limits = */ {
-                /* .nonInductiveForLoops = */ 1,
-                /* .whileLoops = */ 1,
-                /* .doWhileLoops = */ 1,
-                /* .generalUniformIndexing = */ 1,
-                /* .generalAttributeMatrixVectorIndexing = */ 1,
-                /* .generalVaryingIndexing = */ 1,
-                /* .generalSamplerIndexing = */ 1,
-                /* .generalVariableIndexing = */ 1,
-                /* .generalConstantMatrixVectorIndexing = */ 1,
-        }
-};
diff --git a/libs/geometry/src/MikktspaceImpl.cpp b/libs/geometry/src/MikktspaceImpl.cpp
index 5c5a1b5..4c6dd9b 100644
--- a/libs/geometry/src/MikktspaceImpl.cpp
+++ b/libs/geometry/src/MikktspaceImpl.cpp
@@ -21,7 +21,7 @@
 #include <utils/Panic.h>
 
 #include <meshoptimizer.h>
-#include <mikktspace/mikktspace.h>
+#include <mikktspace.h>
 
 #include <vector>
 
diff --git a/libs/imageio/src/BasisEncoder.cpp b/libs/imageio/src/BasisEncoder.cpp
index 0c24e71..fb4eea1 100644
--- a/libs/imageio/src/BasisEncoder.cpp
+++ b/libs/imageio/src/BasisEncoder.cpp
@@ -22,7 +22,7 @@
 
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Warray-bounds"
-#include <basisu_comp.h>
+#include <basisu/encoder/basisu_comp.h>
 #pragma clang diagnostic pop
 
 namespace image {
diff --git a/libs/ktxreader/src/Ktx2Reader.cpp b/libs/ktxreader/src/Ktx2Reader.cpp
index 5d53861..f72c108 100644
--- a/libs/ktxreader/src/Ktx2Reader.cpp
+++ b/libs/ktxreader/src/Ktx2Reader.cpp
@@ -26,7 +26,7 @@
 
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Warray-bounds"
-#include <basisu_transcoder.h>
+#include <basisu/transcoder/basisu_transcoder.h>
 #pragma clang diagnostic pop
 
 using namespace basist;
diff --git a/libs/matdbg/src/ShaderReplacer.cpp b/libs/matdbg/src/ShaderReplacer.cpp
index 9cd8ba4..f388f72 100644
--- a/libs/matdbg/src/ShaderReplacer.cpp
+++ b/libs/matdbg/src/ShaderReplacer.cpp
@@ -27,9 +27,9 @@
 
 #include <sstream>
 
-#include <GlslangToSpv.h>
+#include <glslang/SPIRV/GlslangToSpv.h>
 
-#include "sca/builtinResource.h"
+#include <glslang/Include/ResourceLimits.h>
 #include "sca/GLSLTools.h"
 
 #include "eiff/ChunkContainer.h"
diff --git a/libs/viewer/src/RemoteServer.cpp b/libs/viewer/src/RemoteServer.cpp
index 441c730..89c492f 100644
--- a/libs/viewer/src/RemoteServer.cpp
+++ b/libs/viewer/src/RemoteServer.cpp
@@ -208,8 +208,8 @@ bool MessageReceiver::handleData(CivetServer* server, struct mg_connection* conn
 
 void MessageSender::sendMessage(const char* label, const char* buffer, size_t bufsize) {
     for (auto iter : connections) {
-        mg_websocket_write(iter.first, 0x80, label, strlen(label) + 1);
-        mg_websocket_write(iter.first, 0x80, buffer, bufsize);
+        mg_websocket_write(const_cast<mg_connection*>(iter.first), 0x80, label, strlen(label) + 1);
+        mg_websocket_write(const_cast<mg_connection*>(iter.first), 0x80, buffer, bufsize);
     }
 }
 
diff --git a/samples/frame_generator.cpp b/samples/frame_generator.cpp
index a6a2164..3fb7e80 100644
--- a/samples/frame_generator.cpp
+++ b/samples/frame_generator.cpp
@@ -25,7 +25,7 @@
 #include <utility>
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/samples/gltf_instances.cpp b/samples/gltf_instances.cpp
index 688272d..5210ebd 100644
--- a/samples/gltf_instances.cpp
+++ b/samples/gltf_instances.cpp
@@ -33,7 +33,7 @@
 
 #include <camutils/Manipulator.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/NameComponentManager.h>
 
diff --git a/samples/gltf_viewer.cpp b/samples/gltf_viewer.cpp
index 77aee47..b272d0d 100644
--- a/samples/gltf_viewer.cpp
+++ b/samples/gltf_viewer.cpp
@@ -43,7 +43,7 @@
 
 #include <private/filament/EngineEnums.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/NameComponentManager.h>
 #include <utils/Log.h>
diff --git a/samples/heightfield.cpp b/samples/heightfield.cpp
index 855dabf..92c98ad 100644
--- a/samples/heightfield.cpp
+++ b/samples/heightfield.cpp
@@ -32,7 +32,7 @@
 #include <utils/JobSystem.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <filamentapp/FilamentApp.h>
 
diff --git a/samples/hellomorphing.cpp b/samples/hellomorphing.cpp
index 54ce244..eb10396 100644
--- a/samples/hellomorphing.cpp
+++ b/samples/hellomorphing.cpp
@@ -29,7 +29,7 @@
 #include <utils/EntityManager.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
diff --git a/samples/hellopbr.cpp b/samples/hellopbr.cpp
index f5cae37..08c63d8 100644
--- a/samples/hellopbr.cpp
+++ b/samples/hellopbr.cpp
@@ -29,7 +29,7 @@
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <iostream>
 
diff --git a/samples/helloskinning.cpp b/samples/helloskinning.cpp
index f3d5046..07ee5ab 100644
--- a/samples/helloskinning.cpp
+++ b/samples/helloskinning.cpp
@@ -35,7 +35,7 @@
 #include <cmath>
 #include <iostream>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include "generated/resources/resources.h"
 
diff --git a/samples/helloskinningbuffer.cpp b/samples/helloskinningbuffer.cpp
index 175c62c..59419e3 100644
--- a/samples/helloskinningbuffer.cpp
+++ b/samples/helloskinningbuffer.cpp
@@ -29,7 +29,7 @@
 #include <utils/EntityManager.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
diff --git a/samples/helloskinningbuffer_morebones.cpp b/samples/helloskinningbuffer_morebones.cpp
index 184d799..2f7051c 100644
--- a/samples/helloskinningbuffer_morebones.cpp
+++ b/samples/helloskinningbuffer_morebones.cpp
@@ -29,7 +29,7 @@
 #include <utils/EntityManager.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
diff --git a/samples/hellostereo.cpp b/samples/hellostereo.cpp
index fff1616..ef377d7 100644
--- a/samples/hellostereo.cpp
+++ b/samples/hellostereo.cpp
@@ -37,7 +37,7 @@
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <iostream>
 #include <vector>
diff --git a/samples/hellotriangle.cpp b/samples/hellotriangle.cpp
index 4477429..67959a5 100644
--- a/samples/hellotriangle.cpp
+++ b/samples/hellotriangle.cpp
@@ -31,7 +31,7 @@
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <cmath>
 #include <iostream>
diff --git a/samples/image_viewer.cpp b/samples/image_viewer.cpp
index d380aae..c3266fc 100644
--- a/samples/image_viewer.cpp
+++ b/samples/image_viewer.cpp
@@ -36,7 +36,7 @@
 
 #include <camutils/Manipulator.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <math/half.h>
 #include <math/vec3.h>
diff --git a/samples/lightbulb.cpp b/samples/lightbulb.cpp
index 5c5d735..385a5ac 100644
--- a/samples/lightbulb.cpp
+++ b/samples/lightbulb.cpp
@@ -19,7 +19,7 @@
 #include <map>
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/EntityManager.h>
 #include <utils/Path.h>
diff --git a/samples/material_sandbox.cpp b/samples/material_sandbox.cpp
index 7778246..2c8e881 100644
--- a/samples/material_sandbox.cpp
+++ b/samples/material_sandbox.cpp
@@ -19,7 +19,7 @@
 #include <map>
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <imgui.h>
 
diff --git a/samples/rendertarget.cpp b/samples/rendertarget.cpp
index 9ff53cc..8cf3de9 100644
--- a/samples/rendertarget.cpp
+++ b/samples/rendertarget.cpp
@@ -35,7 +35,7 @@
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <iostream>
 
diff --git a/samples/sample_cloth.cpp b/samples/sample_cloth.cpp
index b7a5627..ec32060 100644
--- a/samples/sample_cloth.cpp
+++ b/samples/sample_cloth.cpp
@@ -19,7 +19,7 @@
 #include <map>
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/samples/sample_full_pbr.cpp b/samples/sample_full_pbr.cpp
index 77c08ec..edc284f 100644
--- a/samples/sample_full_pbr.cpp
+++ b/samples/sample_full_pbr.cpp
@@ -37,7 +37,7 @@
 #include <math/mat4.h>
 #include <math/vec3.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <stb_image.h>
 
diff --git a/samples/sample_normal_map.cpp b/samples/sample_normal_map.cpp
index 684b398..f31aa24 100644
--- a/samples/sample_normal_map.cpp
+++ b/samples/sample_normal_map.cpp
@@ -19,7 +19,7 @@
 #include <map>
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/samples/skinningtest.cpp b/samples/skinningtest.cpp
index a2da0b1..3c173d2 100644
--- a/samples/skinningtest.cpp
+++ b/samples/skinningtest.cpp
@@ -30,7 +30,7 @@
 #include <utils/EntityManager.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <filamentapp/Config.h>
 #include <filamentapp/FilamentApp.h>
diff --git a/samples/suzanne.cpp b/samples/suzanne.cpp
index dae42d3..9e695c7 100644
--- a/samples/suzanne.cpp
+++ b/samples/suzanne.cpp
@@ -35,7 +35,7 @@
 #include <filamentapp/FilamentApp.h>
 #include <filamentapp/IBL.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/samples/viewtest.cpp b/samples/viewtest.cpp
index 26ee9ac..1467589 100644
--- a/samples/viewtest.cpp
+++ b/samples/viewtest.cpp
@@ -26,7 +26,7 @@
 
 #include <utils/EntityManager.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <iostream>
 #include <string>
diff --git a/tools/cmgen/src/cmgen.cpp b/tools/cmgen/src/cmgen.cpp
index 04eb151..c37dff6 100644
--- a/tools/cmgen/src/cmgen.cpp
+++ b/tools/cmgen/src/cmgen.cpp
@@ -44,7 +44,7 @@
 
 #include <string.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 
 using namespace filament::math;
diff --git a/tools/cso-lut/src/main.cpp b/tools/cso-lut/src/main.cpp
index b45a607..0233619 100644
--- a/tools/cso-lut/src/main.cpp
+++ b/tools/cso-lut/src/main.cpp
@@ -31,7 +31,7 @@
 #include <utils/compiler.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 using namespace filament::math;
 using namespace image;
diff --git a/tools/filamesh/src/main.cpp b/tools/filamesh/src/main.cpp
index 541d7ff..e9ba1b5 100644
--- a/tools/filamesh/src/main.cpp
+++ b/tools/filamesh/src/main.cpp
@@ -30,7 +30,7 @@
 
 #include <filameshio/filamesh.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 using namespace filamesh;
 using namespace filament::math;
diff --git a/tools/glslminifier/src/main.cpp b/tools/glslminifier/src/main.cpp
index a6d2f56..4f79833 100644
--- a/tools/glslminifier/src/main.cpp
+++ b/tools/glslminifier/src/main.cpp
@@ -18,7 +18,7 @@
 
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <fstream>
 #include <iostream>
diff --git a/tools/matc/src/matc/CommandlineConfig.cpp b/tools/matc/src/matc/CommandlineConfig.cpp
index 6ca6315..11b90b9 100644
--- a/tools/matc/src/matc/CommandlineConfig.cpp
+++ b/tools/matc/src/matc/CommandlineConfig.cpp
@@ -18,7 +18,7 @@
 
 #include <private/filament/Variant.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/tools/matc/src/matc/MaterialCompiler.cpp b/tools/matc/src/matc/MaterialCompiler.cpp
index 93d0fed..b5f1795 100644
--- a/tools/matc/src/matc/MaterialCompiler.cpp
+++ b/tools/matc/src/matc/MaterialCompiler.cpp
@@ -34,10 +34,9 @@
 #include "JsonishParser.h"
 #include "ParametersProcessor.h"
 
-#include <GlslangToSpv.h>
-#include "glslang/Include/intermediate.h"
-
-#include "sca/builtinResource.h"
+#include <glslang/SPIRV/GlslangToSpv.h>
+#include <glslang/Public/ShaderLang.h>
+#include <glslang/Include/ResourceLimits.h>
 
 #include <smolv.h>
 
diff --git a/tools/matedit/src/main.cpp b/tools/matedit/src/main.cpp
index 50b4791..3632102 100644
--- a/tools/matedit/src/main.cpp
+++ b/tools/matedit/src/main.cpp
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include "ExternalCompile.h"
 
diff --git a/tools/matinfo/src/main.cpp b/tools/matinfo/src/main.cpp
index ea2d1a0..d60f63d 100644
--- a/tools/matinfo/src/main.cpp
+++ b/tools/matinfo/src/main.cpp
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/tools/mipgen/src/main.cpp b/tools/mipgen/src/main.cpp
index c7afabf..8a87dce 100644
--- a/tools/mipgen/src/main.cpp
+++ b/tools/mipgen/src/main.cpp
@@ -26,7 +26,7 @@
 
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <fstream>
 #include <iostream>
diff --git a/tools/normal-blending/src/main.cpp b/tools/normal-blending/src/main.cpp
index e3cadf4..76ec57b 100644
--- a/tools/normal-blending/src/main.cpp
+++ b/tools/normal-blending/src/main.cpp
@@ -25,7 +25,7 @@
 
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 using namespace filament::math;
 using namespace image;
diff --git a/tools/resgen/src/main.cpp b/tools/resgen/src/main.cpp
index 1f475c4..d97aaeb 100644
--- a/tools/resgen/src/main.cpp
+++ b/tools/resgen/src/main.cpp
@@ -16,7 +16,7 @@
 
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <algorithm>
 #include <cstdint>
diff --git a/tools/rgb-to-lmsr/src/main.cpp b/tools/rgb-to-lmsr/src/main.cpp
index 8d4f3cd..07e8e6b 100644
--- a/tools/rgb-to-lmsr/src/main.cpp
+++ b/tools/rgb-to-lmsr/src/main.cpp
@@ -18,7 +18,7 @@
 #include <iomanip>
 #include <iostream>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 
diff --git a/tools/roughness-prefilter/src/main.cpp b/tools/roughness-prefilter/src/main.cpp
index e6ed1d7..ed15d6d 100644
--- a/tools/roughness-prefilter/src/main.cpp
+++ b/tools/roughness-prefilter/src/main.cpp
@@ -33,7 +33,7 @@
 #include <utils/JobSystem.h>
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 using namespace filament::math;
 using namespace image;
diff --git a/tools/specular-color/src/main.cpp b/tools/specular-color/src/main.cpp
index cfbf3c1..8676466 100644
--- a/tools/specular-color/src/main.cpp
+++ b/tools/specular-color/src/main.cpp
@@ -23,7 +23,7 @@
 
 #include <vector>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <utils/Path.h>
 #include <utils/string.h>
diff --git a/tools/uberz/src/main.cpp b/tools/uberz/src/main.cpp
index 43e2005..365222b 100644
--- a/tools/uberz/src/main.cpp
+++ b/tools/uberz/src/main.cpp
@@ -16,7 +16,7 @@
 
 #include <utils/Path.h>
 
-#include <getopt/getopt.h>
+#include <getopt.h>
 
 #include <fstream>
 #include <iostream>
