diff --git a/libs/matdbg/CMakeLists.txt b/libs/matdbg/CMakeLists.txt
index 551196d..f7993e0 100644
--- a/libs/matdbg/CMakeLists.txt
+++ b/libs/matdbg/CMakeLists.txt
@@ -115,10 +115,5 @@ set(MATDBG_DEPS
     spirv-cross-glsl
     )
 
-set(MATDBG_COMBINED_OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/libmatdbg_combined.a")
-# combine_static_libs(matdbg "${MATDBG_COMBINED_OUTPUT}" "${MATDBG_DEPS}")
-
-set(MATDBG_LIB_NAME ${CMAKE_STATIC_LIBRARY_PREFIX}matdbg${CMAKE_STATIC_LIBRARY_SUFFIX})
-install(FILES "${MATDBG_COMBINED_OUTPUT}" DESTINATION lib/${DIST_DIR} RENAME ${MATDBG_LIB_NAME})
-# We do not need matdbg headers in the install directory
-# install(DIRECTORY ${PUBLIC_HDR_DIR}/matdbg DESTINATION include)
+install(TARGETS ${TARGET} DESTINATION ${CMAKE_INSTALL_LIBDIR})
+# install(FILES ${PUBLIC_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/matdbg)
