diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index 58cb402..c940189 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -1144,9 +1144,10 @@ endif()
 if(USE_CUDA)
   find_package(CUB)
   if(NOT CUB_FOUND)
-    message(FATAL_ERROR "Cannot find CUB.")
+    message(WARNING "Cannot find CUB.")
+  else()
+    include_directories(SYSTEM ${CUB_INCLUDE_DIRS})
   endif()
-  include_directories(SYSTEM ${CUB_INCLUDE_DIRS})
 endif()
 
 if(USE_DISTRIBUTED AND USE_TENSORPIPE)
diff --git a/cmake/public/cuda.cmake b/cmake/public/cuda.cmake
index ca888bb..fe9d68c 100644
--- a/cmake/public/cuda.cmake
+++ b/cmake/public/cuda.cmake
@@ -57,6 +57,7 @@ if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.12.0)
 endif()
 
 find_package(CUDAToolkit REQUIRED)
+link_directories("${CUDAToolkit_LIBRARY_DIR}")
 
 cmake_policy(POP)
 
