diff --git a/CMakeLists.txt b/CMakeLists.txt
index f4fe893..489382d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,11 +11,10 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
+cmake_minimum_required(VERSION 3.22)
 cmake_policy(SET CMP0012 NEW)
 cmake_policy(SET CMP0048 NEW)
 project(ruy CXX)
-cmake_minimum_required(VERSION 3.13)  # Copied from IREE
 set(CMAKE_CXX_STANDARD 14)
 
 include(GNUInstallDirs)
diff --git a/ruy/CMakeLists.txt b/ruy/CMakeLists.txt
index 8e493de..fc9bffd 100644
--- a/ruy/CMakeLists.txt
+++ b/ruy/CMakeLists.txt
@@ -2,7 +2,7 @@
 # To regenerate, run:
 #   cmake/bazel_to_cmake.sh
 
-if(CMAKE_SYSTEM_NAME STREQUAL Windows)
+if(CMAKE_SYSTEM_NAME MATCHES Windows) # Windows|WindowsStore
   set(ruy_0_Wall_Wextra_Wundef "")
 else()
   set(ruy_0_Wall_Wextra_Wundef "-Wall;-Wextra;-Wundef")
@@ -105,7 +105,7 @@ ruy_cc_library(
     ${ruy_2_O3}
 )
 
-if(CMAKE_SYSTEM_NAME STREQUAL Windows)
+if(CMAKE_SYSTEM_NAME MATCHES Windows) # Windows|WindowsStore
   set(ruy_3_pthread "")
 else()
   set(ruy_3_pthread "-pthread")
@@ -393,7 +393,7 @@ ruy_cc_library(
     ${ruy_2_O3}
 )
 
-if(CMAKE_SYSTEM_NAME STREQUAL Windows)
+if(CMAKE_SYSTEM_NAME MATCHES Windows) # Windows|WindowsStore
   set(ruy_4_Wno_undef "")
 else()
   set(ruy_4_Wno_undef "-Wno-undef")
@@ -1297,7 +1297,7 @@ ruy_cc_library(
     ruy_mul_params
 )
 
-if(CMAKE_SYSTEM_NAME STREQUAL Windows)
+if(CMAKE_SYSTEM_NAME MATCHES Windows) # Windows|WindowsStore
   set(ruy_10_lm "")
 else()
   set(ruy_10_lm "-lm")
diff --git a/ruy/profiler/CMakeLists.txt b/ruy/profiler/CMakeLists.txt
index df4b30a..ddbc886 100644
--- a/ruy/profiler/CMakeLists.txt
+++ b/ruy/profiler/CMakeLists.txt
@@ -8,7 +8,7 @@ else()
   set(ruy_profiler_0_RUY_PROFILER "")
 endif()
 
-if(CMAKE_SYSTEM_NAME STREQUAL Windows)
+if(CMAKE_SYSTEM_NAME MATCHES Windows) # Windows|WindowsStore
   set(ruy_profiler_1_pthread "")
 else()
   set(ruy_profiler_1_pthread "-pthread")
