diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9831cee..a7b477e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,7 @@ cmake_minimum_required(VERSION 3.15.1)
 
 project(SwiftCrypto
   LANGUAGES ASM C Swift)
+include(GNUInstallDirs)
 
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
 
@@ -37,13 +38,13 @@ if(BUILD_SHARED_LIBS)
   set(CMAKE_POSITION_INDEPENDENT_CODE YES)
 endif()
 
-if(CMAKE_SYSTEM_NAME STREQUAL Darwin AND NOT CMAKE_CROSSCOMPILING)
+if(FALSE)
   set(CMAKE_AR "/usr/bin/ar")
   set(CMAKE_C_ARCHIVE_CREATE "<CMAKE_AR> crs <TARGET> <LINK_FLAGS> <OBJECTS>")
   set(CMAKE_RANLIB "/usr/bin/ranlib")
 endif()
 
-if(NOT CMAKE_SYSTEM_NAME STREQUAL Darwin)
+if(NOT APPLE)
   find_package(dispatch CONFIG)
   find_package(Foundation CONFIG)
 endif()
diff --git a/Sources/CCryptoBoringSSL/CMakeLists.txt b/Sources/CCryptoBoringSSL/CMakeLists.txt
index ddf4786..29577a4 100644
--- a/Sources/CCryptoBoringSSL/CMakeLists.txt
+++ b/Sources/CCryptoBoringSSL/CMakeLists.txt
@@ -12,7 +12,8 @@
 ##
 ##===----------------------------------------------------------------------===##
 
-add_library(CCryptoBoringSSL STATIC
+file(GLOB headers ${CMAKE_CURRENT_SOURCE_DIR}/include/*.h)
+add_library(CCryptoBoringSSL STATIC ${headers}
   "crypto/asn1/a_bitstr.c"
   "crypto/asn1/a_bool.c"
   "crypto/asn1/a_d2i_fp.c"
@@ -397,6 +398,11 @@ target_include_directories(CCryptoBoringSSL PUBLIC
 target_compile_definitions(CCryptoBoringSSL PRIVATE
   $<$<PLATFORM_ID:Windows>:WIN32_LEAN_AND_MEAN>)
 set_target_properties(CCryptoBoringSSL PROPERTIES
+  PUBLIC_HEADER "${headers}"
+  FRAMEWORK true
   INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_SOURCE_DIR}/include;${CMAKE_Swift_MODULE_DIRECTORY}")
 
 set_property(GLOBAL APPEND PROPERTY SWIFT_CRYPTO_EXPORTS CCryptoBoringSSL)
+
+install(FILES ${headers} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/CCryptoBoringSSL)
+install(TARGETS CCryptoBoringSSL FRAMEWORK DESTINATION ${CMAKE_INSTALL_LIBDIR})
\ No newline at end of file
diff --git a/Sources/CCryptoBoringSSLShims/CMakeLists.txt b/Sources/CCryptoBoringSSLShims/CMakeLists.txt
index 5550b15..bcabdb8 100644
--- a/Sources/CCryptoBoringSSLShims/CMakeLists.txt
+++ b/Sources/CCryptoBoringSSLShims/CMakeLists.txt
@@ -12,7 +12,8 @@
 ##
 ##===----------------------------------------------------------------------===##
 
-add_library(CCryptoBoringSSLShims STATIC
+file(GLOB headers ${CMAKE_CURRENT_SOURCE_DIR}/include/*.h)
+add_library(CCryptoBoringSSLShims STATIC ${headers}
   "shims.c")
 
 target_include_directories(CCryptoBoringSSLShims PUBLIC
@@ -23,6 +24,10 @@ target_link_libraries(CCryptoBoringSSLShims PUBLIC
   CCryptoBoringSSL)
 
 set_target_properties(CCryptoBoringSSLShims PROPERTIES
+  PUBLIC_HEADER "${headers}"
+  FRAMEWORK true
   INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_Swift_MODULE_DIRECTORY})
 
 set_property(GLOBAL APPEND PROPERTY SWIFT_CRYPTO_EXPORTS CCryptoBoringSSLShims)
+install(FILES ${headers} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/CCryptoBoringSSLShims)
+install(TARGETS CCryptoBoringSSLShims FRAMEWORK DESTINATION ${CMAKE_INSTALL_LIBDIR})
diff --git a/cmake/modules/SwiftSupport.cmake b/cmake/modules/SwiftSupport.cmake
index c11d56d..c62bb38 100644
--- a/cmake/modules/SwiftSupport.cmake
+++ b/cmake/modules/SwiftSupport.cmake
@@ -81,9 +81,10 @@ function(_install_target module)
   endif()
 
   install(TARGETS ${module}
-    ARCHIVE DESTINATION lib/${swift}/${swift_os}
-    LIBRARY DESTINATION lib/${swift}/${swift_os}
-    RUNTIME DESTINATION bin)
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  )
   if(type STREQUAL EXECUTABLE)
     return()
   endif()
@@ -97,14 +98,14 @@ function(_install_target module)
   if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
     install(FILES $<TARGET_PROPERTY:${module},Swift_MODULE_DIRECTORY>/${module_name}.swiftdoc
       DESTINATION lib/${swift}/${swift_os}/${module_name}.swiftmodule
-      RENAME ${swift_arch}.swiftdoc)
+      RENAME ${swift_arch}.swiftdoc OPTIONAL)
     install(FILES $<TARGET_PROPERTY:${module},Swift_MODULE_DIRECTORY>/${module_name}.swiftmodule
       DESTINATION lib/${swift}/${swift_os}/${module_name}.swiftmodule
-      RENAME ${swift_arch}.swiftmodule)
+      RENAME ${swift_arch}.swiftmodule OPTIONAL)
   else()
     install(FILES
       $<TARGET_PROPERTY:${module},Swift_MODULE_DIRECTORY>/${module_name}.swiftdoc
       $<TARGET_PROPERTY:${module},Swift_MODULE_DIRECTORY>/${module_name}.swiftmodule
-      DESTINATION lib/${swift}/${swift_os}/${swift_arch})
+      DESTINATION lib/${swift}/${swift_os}/${swift_arch} OPTIONAL)
   endif()
 endfunction()
