diff --git a/protobuf/CMakeLists.txt b/protobuf/CMakeLists.txt
index 34f8374..ef6a428 100644
--- a/protobuf/CMakeLists.txt
+++ b/protobuf/CMakeLists.txt
@@ -53,7 +53,7 @@ if(${TRITON_COMMON_ENABLE_PROTOBUF})
   endif()
 
   add_library(
-    proto-library EXCLUDE_FROM_ALL OBJECT
+    proto-library STATIC
     ${PROTO_SRCS} ${PROTO_HDRS}
   )
 
@@ -61,6 +61,7 @@ if(${TRITON_COMMON_ENABLE_PROTOBUF})
     proto-library
     PUBLIC
       $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+      $<INSTALL_INTERFACE:include>
   )
 
   target_link_libraries(
@@ -125,7 +126,7 @@ if(${TRITON_COMMON_ENABLE_GRPC})
   endif()
 
   add_library(
-    grpc-service-library EXCLUDE_FROM_ALL OBJECT
+    grpc-service-library STATIC
     ${GRPC_SRCS} ${GRPC_HDRS}
   )
 
@@ -133,6 +134,7 @@ if(${TRITON_COMMON_ENABLE_GRPC})
     grpc-service-library
     PUBLIC
       $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+      $<INSTALL_INTERFACE:include>
   )
 
   target_link_libraries(
@@ -155,7 +157,6 @@ if(${TRITON_COMMON_ENABLE_GRPC})
     FILES
       ${CMAKE_CURRENT_BINARY_DIR}/grpc_service.grpc.pb.h
     DESTINATION include
-    OPTIONAL
   )
 endif()
 
@@ -181,7 +182,7 @@ if(${TRITON_COMMON_ENABLE_GRPC})
   )
 
   add_library(
-    grpc-health-library EXCLUDE_FROM_ALL OBJECT
+    grpc-health-library STATIC
     ${HEALTH_SRCS} ${HEALTH_HDRS}
   )
 
@@ -189,6 +190,7 @@ if(${TRITON_COMMON_ENABLE_GRPC})
     grpc-health-library
     PUBLIC
       $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+      $<INSTALL_INTERFACE:include>
   )
 
   target_link_libraries(
@@ -207,6 +209,5 @@ if(${TRITON_COMMON_ENABLE_GRPC})
     FILES
       ${CMAKE_CURRENT_BINARY_DIR}/health.grpc.pb.h
     DESTINATION include
-    OPTIONAL
   )
 endif()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4c98094..77a5913 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -201,4 +201,3 @@ set_target_properties(
     OUTPUT_NAME tritontableprinter
 )
 
-add_subdirectory(test)
