diff --git a/libkineto/CMakeLists.txt b/libkineto/CMakeLists.txt
index 5256694..bc47d54 100644
--- a/libkineto/CMakeLists.txt
+++ b/libkineto/CMakeLists.txt
@@ -135,6 +135,14 @@ IF (DEFINED LIBKINETO_NOXPUPTI AND NOT LIBKINETO_NOXPUPTI)
     add_subdirectory(src/plugin/xpupti)
 endif()
 
+if(NOT LIBKINETO_NOCUPTI)
+  find_package(CUDAToolkit REQUIRED)
+  math(EXPR CUDA_VERSION "${CUDAToolkit_VERSION_MAJOR}000 + ${CUDAToolkit_VERSION_MINOR}0" OUTPUT_FORMAT DECIMAL)
+  add_compile_definitions(CUDA_VERSION=${CUDA_VERSION})
+  get_target_property(CUPTI_INCLUDE_DIR CUDA::cupti INTERFACE_INCLUDE_DIRECTORIES)
+  get_target_property(CUDA_cupti_LIBRARY CUDA::cupti IMPORTED_LOCATION)
+endif()
+
 # Define file lists
 if (LIBKINETO_NOCUPTI AND LIBKINETO_NOROCTRACER AND LIBKINETO_NOXPUPTI)
     get_filelist("get_libkineto_cpu_only_srcs(with_api=False)" LIBKINETO_SRCS)
@@ -213,6 +221,7 @@ set(DYNOLOG_INCLUDE_DIR "${LIBKINETO_THIRDPARTY_DIR}/dynolog/")
 set(IPCFABRIC_INCLUDE_DIR "${DYNOLOG_INCLUDE_DIR}/dynolog/src/ipcfabric/")
 add_subdirectory("${IPCFABRIC_INCLUDE_DIR}")
 
+find_package(fmt CONFIG REQUIRED)
 if(NOT TARGET fmt::fmt-header-only)
   if(NOT FMT_SOURCE_DIR)
     set(FMT_SOURCE_DIR "${LIBKINETO_THIRDPARTY_DIR}/fmt"
