diff --git a/tensorflow/lite/CMakeLists.txt b/tensorflow/lite/CMakeLists.txt
index 38f94e3a..15fdb224 100644
--- a/tensorflow/lite/CMakeLists.txt
+++ b/tensorflow/lite/CMakeLists.txt
@@ -505,9 +505,17 @@ if(_TFLITE_ENABLE_NNAPI)
   )
 
   list(APPEND TFLITE_NNAPI_SRCS
+    "${TF_SOURCE_DIR}/tsl/platform/default/logging.cc"
+    "${TF_SOURCE_DIR}/tsl/platform/default/mutex.cc"
     "${TFLITE_SOURCE_DIR}/nnapi/sl/SupportLibrary.cc"
   )
-
+  # tsl::mutex requires nsync_cpp
+  find_library(NSYNC_LIBRARY NAMES nsync_cpp REQUIRED)
+  list(APPEND TFLITE_TARGET_DEPENDENCIES ${NSYNC_LIBRARY})
+  # disable ml_dtypes/float8.h which requries libeigen3 master branch ...
+  set_source_files_properties(${TFLITE_NNAPI_SRCS} PROPERTIES
+    COMPILE_DEFINITIONS "TENSORFLOW_TSL_PLATFORM_FLOAT8_H_;TENSORFLOW_CORE_PLATFORM_FLOAT8_H_"
+  )
   if(${TFLITE_ENABLE_NNAPI_VERBOSE_VALIDATION})
     list(APPEND TFLITE_TARGET_PUBLIC_OPTIONS "-DNNAPI_VERBOSE_VALIDATION")
   endif()
