diff --git a/tensorflow/lite/CMakeLists.txt b/tensorflow/lite/CMakeLists.txt
index b873c502..3a40a99b 100644
--- a/tensorflow/lite/CMakeLists.txt
+++ b/tensorflow/lite/CMakeLists.txt
@@ -217,7 +217,7 @@ endif()
 # Additional compiler flags used when compiling TF Lite.
 set(TFLITE_TARGET_PUBLIC_OPTIONS "-DEIGEN_NEON_GEBP_NR=4")
 set(TFLITE_TARGET_PRIVATE_OPTIONS "")
-set(TFLITE_TARGET_PRIVATE_DEFINITIONS "")
+set(TFLITE_TARGET_PRIVATE_DEFINITIONS -DTF_MAJOR_VERSION=2 -DTF_MINOR_VERSION=20 -DTF_PATCH_VERSION=0 -DTF_VERSION_SUFFIX="-vcpkg")
 if(CMAKE_CXX_COMPILER_ID MATCHES "Clang$")
   # TFLite uses deprecated methods in neon2sse which generates a huge number of
   # warnings so surpress these until they're fixed.
@@ -794,7 +794,8 @@ set_target_properties(tensorflow-lite PROPERTIES OUTPUT_NAME ${TFLITE_C_OUTPUT_N
 if (BUILD_SHARED_LIBS)
   if (WIN32)
     target_compile_definitions(tensorflow-lite PRIVATE TFL_COMPILE_LIBRARY)
-    target_compile_definitions(xnnpack-delegate PRIVATE TFL_COMPILE_LIBRARY)
+    target_compile_definitions(xnnpack-delegate PRIVATE TFL_COMPILE_LIBRARY ${TFLITE_TARGET_PRIVATE_DEFINITIONS})
+    target_compile_options(xnnpack-delegate PRIVATE ${TFLITE_TARGET_PRIVATE_OPTIONS})
   elseif (APPLE)
     target_link_options(tensorflow-lite PRIVATE "-Wl,-exported_symbols_list,${TFLITE_SOURCE_DIR}/c/exported_symbols.lds")
   else ()
