diff --git a/tensorflow/lite/CMakeLists.txt b/tensorflow/lite/CMakeLists.txt
index 161e3421..102131d4 100644
--- a/tensorflow/lite/CMakeLists.txt
+++ b/tensorflow/lite/CMakeLists.txt
@@ -26,7 +26,7 @@
 # - Many features in experimental
 # - Host Tools (i.e conversion / analysis tools etc.)
 
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.27)
 if(NOT CMAKE_BUILD_TYPE)
   message(STATUS "Setting build type to Release, for debug builds use"
     "'-DCMAKE_BUILD_TYPE=Debug'.")
@@ -268,18 +268,46 @@ populate_tflite_source_vars("core/tools" TFLITE_CORE_TOOLS_SRCS)
 populate_tflite_source_vars("c" TFLITE_C_SRCS)
 populate_tflite_source_vars("delegates" TFLITE_DELEGATES_SRCS)
 if(TFLITE_ENABLE_GPU)
-  find_package(opencl_headers REQUIRED)
-  find_package(vulkan_headers REQUIRED)
-  find_package(fp16_headers REQUIRED)
-  # Android NDK already has OpenGL, EGL headers.
-  if(NOT "${CMAKE_SYSTEM_NAME}" STREQUAL "Android")
-    find_package(opengl_headers REQUIRED)
-    find_package(egl_headers REQUIRED)
-  endif()
-  populate_tflite_source_vars(
-    "delegates/gpu/cl" TFLITE_DELEGATES_GPU_CL_SRCS
-    FILTER "(_test|gl_interop|gpu_api_delegate|egl_sync)\\.(cc|h)$"
-  )
+  find_package(OpenCL REQUIRED) # OpenCL::OpenCL
+  find_package(Vulkan COMPONENTS Headers) # Vulkan::Headers Vulkan::Vulkan
+  if(Vulkan_FOUND)
+    list(APPEND TFLITE_TARGET_DEPENDENCIES Vulkan::Headers)
+  endif()
+  if(ANDROID OR WIN32)
+    set(OPENGL_USE_EGL true)
+    find_path(OPENGL_EGL_INCLUDE_DIRS "EGL/egl.h" REQUIRED)
+    find_library(OPENGL_egl_LIBRARY NAMES EGL libEGL REQUIRED)
+    find_path(OPENGL_GLES3_INCLUDE_DIR "GLES3/gl3.h" REQUIRED)
+    find_library(OPENGL_gles3_LIBRARY NAMES GLESv3 libGLESv3 GLESv2 libGLESv2 REQUIRED)
+    find_library(OPENGL_gl_LIBRARY NAMES GLESv3 libGLESv3 GLESv2 libGLESv2 REQUIRED)
+    # it's using "version 310 es"
+    find_package(OpenGL REQUIRED COMPONENTS GLES3) # OpenGL::GLES3
+    list(APPEND TFLITE_TARGET_DEPENDENCIES OpenGL::GLES3 ${OPENGL_egl_LIBRARY})
+    populate_tflite_source_vars("delegates/gpu/gl" TFLITE_DELEGATES_GPU_GL_SRCS FILTER "(_test)\\.(cc|h)$")
+    populate_tflite_source_vars("delegates/gpu/gl/compiler" TFLITE_DELEGATES_GPU_GL_COMPILER_SRCS FILTER "(_test)\\.(cc|h)$")
+    populate_tflite_source_vars("delegates/gpu/gl/converters" TFLITE_DELEGATES_GPU_GL_CONVERTERS_SRCS FILTER "(_test)\\.(cc|h)$")
+    populate_tflite_source_vars("delegates/gpu/gl/kernels" TFLITE_DELEGATES_GPU_GL_KERNELS_SRCS FILTER "(_test)\\.(cc|h)$")
+    populate_tflite_source_vars("delegates/gpu/gl/workgroups" TFLITE_DELEGATES_GPU_GL_WORKGROUPS_SRCS FILTER "(_test)\\.(cc|h)$")
+    list(APPEND TFLITE_DELEGATES_GPU_SRCS
+      ${TFLITE_DELEGATES_GPU_GL_SRCS}
+      ${TFLITE_DELEGATES_GPU_GL_COMPILER_SRCS}
+      ${TFLITE_DELEGATES_GPU_GL_CONVERTERS_SRCS}
+      ${TFLITE_DELEGATES_GPU_GL_KERNELS_SRCS}
+      ${TFLITE_DELEGATES_GPU_GL_WORKGROUPS_SRCS}
+    )
+  endif()
+  if(ANDROID OR WIN32) # import EGL/OpenGL sources
+    populate_tflite_source_vars(
+      "delegates/gpu/cl" TFLITE_DELEGATES_GPU_CL_SRCS
+      FILTER "(_test)\\.(cc|h)$"
+    )
+  else()
+    populate_tflite_source_vars(
+      "delegates/gpu/cl" TFLITE_DELEGATES_GPU_CL_SRCS
+      FILTER "(_test|gl_interop|gpu_api_delegate|egl_sync)\\.(cc|h)$"
+    )
+    list(APPEND TFLITE_TARGET_PRIVATE_OPTIONS "-DCL_DELEGATE_NO_GL")
+  endif()
   populate_tflite_source_vars(
     "delegates/gpu/cl/default" TFLITE_DELEGATES_GPU_CL_DEFAULT_SRCS
     FILTER "(_test)\\.(cc|h)$"
@@ -355,11 +383,12 @@ if(TFLITE_ENABLE_GPU)
     ${TFLITE_SOURCE_DIR}/delegates/gpu/common
     ${TFLITE_SOURCE_DIR}/delegates/gpu/common/task
   )
-  if(TFLITE_ENABLE_METAL AND "${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
+  if(TFLITE_ENABLE_METAL AND APPLE)
     #
     # libmetal_delegate library
     #
     enable_language(OBJCXX)
+    add_compile_options(-fobjc-arc)
     list(APPEND TFLITE_DELEGATES_METAL_SRCS
       ${TFLITE_SOURCE_DIR}/delegates/gpu/metal_delegate.mm
       ${TFLITE_SOURCE_DIR}/delegates/gpu/metal/buffer.cc
@@ -375,30 +404,16 @@ if(TFLITE_ENABLE_GPU)
       ${TFLITE_DELEGATES_METAL_SRCS}
     )
     target_include_directories(metal_delegate PUBLIC
-      ${CMAKE_BINARY_DIR}/abseil-cpp
-      ${CMAKE_BINARY_DIR}/flatbuffers/include
       PRIVATE ${TENSORFLOW_SOURCE_DIR}
     )
+    target_link_libraries(metal_delegate PUBLIC
+      absl::status
+      flatbuffers::flatbuffers
+    )
     #
     # generate flatbuffers header for inference_context
     #
-    if(FLATBUFFERS_FLATC_EXECUTABLE)
-      set(FLATC ${FLATBUFFERS_FLATC_EXECUTABLE})
-    else()
-      set(FLATC flatc)
-    endif()
-    add_custom_command(
-        OUTPUT ${TFLITE_SOURCE_DIR}/delegates/gpu/metal/inference_context_generated.h
-        COMMAND ${FLATC} --scoped-enums
-        -I ${TENSORFLOW_SOURCE_DIR}
-        -o ${TFLITE_SOURCE_DIR}/delegates/gpu/metal
-        -c ${TFLITE_SOURCE_DIR}/delegates/gpu/metal/inference_context.fbs
-    )
-    add_custom_target(
-        inference_context_cc_fbs
-        DEPENDS ${TFLITE_SOURCE_DIR}/delegates/gpu/metal/inference_context_generated.h
-    )
-    add_dependencies(metal_delegate inference_context_cc_fbs)
+    # note: it's already generated in portfile.cmake
     #
     # supplementary libraries for libmetal_delegate
     #
@@ -415,12 +430,13 @@ if(TFLITE_ENABLE_GPU)
    foreach(lib_name ${CC_SRCS})
      set_source_files_properties(${METAL_DELEGATE_PATH}${lib_name}.cc  PROPERTIES LANGUAGE OBJCXX)
      add_library("${lib_name}" STATIC ${METAL_DELEGATE_PATH}${lib_name}.cc)
-     target_include_directories("${lib_name}" PUBLIC
-       ${CMAKE_BINARY_DIR}/abseil-cpp
-       ${CMAKE_BINARY_DIR}/flatbuffers/include
+     target_include_directories("${lib_name}" PUBLIC ${TENSORFLOW_SOURCE_DIR})
+     target_link_libraries("${lib_name}" PUBLIC
+       absl::base
+       flatbuffers::flatbuffers
      )
      set_target_properties(${lib_name} PROPERTIES LINKER_LANGUAGE OBJCXX)
-     target_link_libraries(${lib_name})
+     list(APPEND TFLITE_TARGET_DEPENDENCIES "${lib_name}")
    endforeach()
 
    list(APPEND MM_SRCS
@@ -429,15 +445,23 @@ if(TFLITE_ENABLE_GPU)
    )
    foreach(lib_name ${MM_SRCS})
      add_library("${lib_name}" STATIC ${METAL_DELEGATE_PATH}${lib_name}.mm)
-     target_include_directories("${lib_name}" PUBLIC
-       ${CMAKE_BINARY_DIR}/abseil-cpp
-       ${CMAKE_BINARY_DIR}/flatbuffers/include
+     target_include_directories("${lib_name}" PUBLIC ${TENSORFLOW_SOURCE_DIR})
+     target_link_libraries("${lib_name}" PUBLIC
+       absl::base
+       flatbuffers::flatbuffers
      )
-     target_link_libraries(${lib_name})
+     list(APPEND TFLITE_TARGET_DEPENDENCIES "${lib_name}")
    endforeach()
+   list(APPEND TFLITE_TARGET_DEPENDENCIES metal_delegate "-framework Metal" "-framework Foundation")
 endif()
-  list(APPEND TFLITE_TARGET_PUBLIC_OPTIONS "-DCL_DELEGATE_NO_GL" "-DEGL_NO_X11")
+  message(STATUS "using CL_TARGET_OPENCL_VERSION: ${OpenCL_VERSION_MAJOR}${OpenCL_VERSION_MINOR}0")
+  list(APPEND TFLITE_TARGET_PUBLIC_OPTIONS
+    "-DEGL_EGLEXT_PROTOTYPES"
+    "-DCL_TARGET_OPENCL_VERSION=${OpenCL_VERSION_MAJOR}${OpenCL_VERSION_MINOR}0"
+    "-DEGL_NO_X11"
+  )
   list(APPEND TFLITE_TARGET_DEPENDENCIES
+    OpenCL::OpenCL
     absl::any
     absl::flat_hash_map
   )
@@ -620,6 +644,7 @@ target_link_libraries(tensorflow-lite
     ruy::ruy
     ${PTHREADPOOL_LIB}
     ${CMAKE_DL_LIBS}
+  PRIVATE
     ${TFLITE_TARGET_DEPENDENCIES}
 )
 
