diff --git a/tensorflow/lite/delegates/telemetry.h b/tensorflow/lite/delegates/telemetry.h
index d7e92be5..6affa9ae 100644
--- a/tensorflow/lite/delegates/telemetry.h
+++ b/tensorflow/lite/delegates/telemetry.h
@@ -90,6 +90,8 @@ class DelegateStatus {
   int32_t code_;
 };
 
+using tflite::proto::TFLiteSettings;
+
 // Used by delegates to report their configuration/settings to TFLite.
 // Calling this method adds a new GENERAL_RUNTIME_INSTRUMENTATION_EVENT to
 // the runtime Profiler.
diff --git a/tensorflow/lite/kernels/random_ops.cc b/tensorflow/lite/kernels/random_ops.cc
index 8636ca20..8f8eb564 100644
--- a/tensorflow/lite/kernels/random_ops.cc
+++ b/tensorflow/lite/kernels/random_ops.cc
@@ -17,6 +17,7 @@ limitations under the License.
 #include <cstdint>
 #include <random>
 
+#include "tensorflow/lite/kernels/internal/constants.h"
 #include "tensorflow/core/lib/random/philox_random.h"
 #include "tensorflow/core/lib/random/random_distributions_utils.h"
 #include "tensorflow/lite/c/builtin_op_data.h"
diff --git a/tensorflow/lite/simple_memory_arena.cc b/tensorflow/lite/simple_memory_arena.cc
index 1c7a0384..59a139ec 100644
--- a/tensorflow/lite/simple_memory_arena.cc
+++ b/tensorflow/lite/simple_memory_arena.cc
@@ -168,11 +168,17 @@ TfLiteStatus SimpleMemoryArena::ReleaseBuffer() {
   return kTfLiteOk;
 }
 
+#if defined(_DEBUG)
+void DumpArenaInfo(const std::string& name,
+                   const std::vector<int>& execution_plan, size_t arena_size,
+                   const std::vector<ArenaAllocWithUsageInterval>& allocs);
+#else
 // Using weak symbols to create a pluggable debugging module.
 TFLITE_ATTRIBUTE_WEAK void DumpArenaInfo(
     const std::string& name, const std::vector<int>& execution_plan,
     size_t arena_size, const std::vector<ArenaAllocWithUsageInterval>& allocs) {
 }
+#endif
 
 void SimpleMemoryArena::DumpDebugInfo(
     const std::string& name, const std::vector<int>& execution_plan) const {
diff --git a/tensorflow/lite/simple_memory_arena_debug_dump.cc b/tensorflow/lite/simple_memory_arena_debug_dump.cc
index 0cf80051..26547a0c 100644
--- a/tensorflow/lite/simple_memory_arena_debug_dump.cc
+++ b/tensorflow/lite/simple_memory_arena_debug_dump.cc
@@ -129,6 +129,7 @@ class TopKLayers {
 };
 }  // namespace
 
+#if defined(_DEBUG)
 // Corresponding weak declaration found in lite/simple_memory_arena.cc
 void DumpArenaInfo(const std::string& name,
                    const std::vector<int>& execution_plan, size_t arena_size,
@@ -194,4 +195,5 @@ void DumpArenaInfo(const std::string& name,
   top_usage.Print();
   printf("===End of %s ===\n\n", name.c_str());
 }
+#endif
 }  // namespace tflite
diff --git a/tensorflow/lite/delegates/gpu/gl/runtime.cc b/tensorflow/lite/delegates/gpu/gl/runtime.cc
index fe3db954..edee334b 100644
--- a/tensorflow/lite/delegates/gpu/gl/runtime.cc
+++ b/tensorflow/lite/delegates/gpu/gl/runtime.cc
@@ -90,27 +90,22 @@ absl::Status MakeGlTexture(const Object& object, const ObjectData& data,
       if (data.size() % 2 != 0) {
         return absl::InvalidArgumentError("Texture size is not aligned");
       }
-      return std::visit(
-          TextureF16Maker{
-              .data = absl::MakeConstSpan(
-                  reinterpret_cast<const uint16_t*>(data.data()),
-                  data.size() / 2),
-              .gl_texture = gl_texture,
-          },
-          object.size);
+      TextureF16Maker marker{};
+      marker.data = absl::MakeConstSpan(
+          reinterpret_cast<const uint16_t*>(data.data()), data.size() / 2);
+      marker.gl_texture = gl_texture;
+      return std::visit(marker, object.size);
     }
     case DataType::FLOAT32: {
       if (data.size() % sizeof(float) != 0) {
         return absl::InvalidArgumentError("Texture size is not aligned");
       }
-      return std::visit(
-          TextureF32Maker{
-              .data = absl::MakeConstSpan(
-                  reinterpret_cast<const float*>(data.data()),
-                  data.size() / sizeof(float)),
-              .gl_texture = gl_texture,
-          },
-          object.size);
+      TextureF32Maker marker{};
+      marker.data =
+          absl::MakeConstSpan(reinterpret_cast<const float*>(data.data()),
+                              data.size() / sizeof(float));
+      marker.gl_texture = gl_texture;
+      return std::visit(marker, object.size);
     }
     default:
       return absl::InvalidArgumentError("Unsupported textures data type.");
