diff --git a/tensorflow/lite/CMakeLists.txt b/tensorflow/lite/CMakeLists.txt
index 3ba38af5..a8d15fcb 100644
--- a/tensorflow/lite/CMakeLists.txt
+++ b/tensorflow/lite/CMakeLists.txt
@@ -197,7 +197,8 @@ if(CMAKE_SYSTEM_NAME MATCHES "Windows")
   endif()
 endif()
 if(CMAKE_SYSTEM_NAME MATCHES "Android")
-  find_library(ANDROID_LOG_LIB log)
+  find_library(ANDROID_LOG_LIB log REQUIRED)
+  find_library(NSYNC_LIB NAMES nsync_cpp nsync REQUIRED)
 endif()
 # Build a list of source files to compile into the TF Lite library.
 populate_tflite_source_vars("." TFLITE_SRCS)
@@ -216,6 +217,10 @@ else()
 endif()
 if(ANDROID)
   list(FILTER TFLITE_SRCS EXCLUDE REGEX ".*minimal_logging_default\\.cc$")
+  list(APPEND TFLITE_SRCS
+    ${TF_SOURCE_DIR}/core/platform/default/logging.cc
+    ${TF_SOURCE_DIR}/core/platform/default/mutex.cc
+  )
 else()
   list(FILTER TFLITE_SRCS EXCLUDE REGEX ".*minimal_logging_android\\.cc$")
 endif()
@@ -643,6 +648,7 @@ target_link_libraries(tensorflow-lite
     unofficial::ruy
     ${CMAKE_DL_LIBS}
     ${TFLITE_TARGET_DEPENDENCIES}
+    ${ANDROID_LOG_LIB} ${NSYNC_LIB}
 )
 
 if (NOT BUILD_SHARED_LIBS)
