diff --git a/tensorflow/lite/CMakeLists.txt b/tensorflow/lite/CMakeLists.txt
index 635e4a80..729e5b36 100644
--- a/tensorflow/lite/CMakeLists.txt
+++ b/tensorflow/lite/CMakeLists.txt
@@ -750,6 +750,39 @@ target_compile_definitions(tensorflow-lite
 )
 add_library(${PROJECT_NAME}::tensorflowlite ALIAS tensorflow-lite)
 
+# see tensorflow/lite/c/CMakeLists.txt
+target_sources(tensorflow-lite PRIVATE
+  ${TFLITE_SOURCE_DIR}/core/c/c_api.cc
+  ${TFLITE_SOURCE_DIR}/core/c/c_api_experimental.cc
+  ${TFLITE_SOURCE_DIR}/core/c/common.cc
+  ${TFLITE_SOURCE_DIR}/core/c/operator.cc
+  c/builtin_op_data.h
+  c/c_api.h
+  c/c_api_experimental.h
+  c/c_api_internal.h
+  c/c_api_types.h
+  c/common.h
+  c/c_api_opaque_internal.cc
+  c/c_api_opaque_internal.h
+  c/c_api_opaque.h
+  c/common_internal.cc
+  c/common_internal.h
+)
+target_include_directories(tensorflow-lite PRIVATE c)
+# set_target_properties(tensorflow-lite PROPERTEIS
+#   OUTPUT_NAME tensorflowlite_c
+# )
+
+if (BUILD_SHARED_LIBS)
+  if (WIN32)
+    target_compile_definitions(tensorflow-lite PRIVATE TFL_COMPILE_LIBRARY)
+  elseif (APPLE)
+    target_link_options(tensorflow-lite PRIVATE "-Wl,-exported_symbols_list,${TFLITE_SOURCE_DIR}/c/exported_symbols.lds")
+  else ()
+    target_link_options(tensorflow-lite PRIVATE "-Wl,--version-script,${TFLITE_SOURCE_DIR}/c/version_script.lds")
+  endif()
+endif()
+
 # The install targets.
 if(TFLITE_ENABLE_INSTALL)
   install(
@@ -757,6 +790,7 @@ if(TFLITE_ENABLE_INSTALL)
     EXPORT ${PROJECT_NAME}Targets
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
   )
 
   foreach(hdr ${_ALL_TFLITE_HDRS})
