diff --git a/third_party/xla/third_party/tsl/tsl/platform/ml_dtypes.h b/third_party/xla/third_party/tsl/tsl/platform/ml_dtypes.h
index c25efc2f..23a04428 100644
--- a/third_party/xla/third_party/tsl/tsl/platform/ml_dtypes.h
+++ b/third_party/xla/third_party/tsl/tsl/platform/ml_dtypes.h
@@ -16,8 +16,8 @@ limitations under the License.
 #ifndef TENSORFLOW_TSL_PLATFORM_ML_DTYPES_H_
 #define TENSORFLOW_TSL_PLATFORM_ML_DTYPES_H_
 
-#include "ml_dtypes/include/float8.h"  // from @ml_dtypes
-#include "ml_dtypes/include/int4.h"  // from @ml_dtypes
+#include "ml_dtypes/float8.h"  // from @ml_dtypes
+#include "ml_dtypes/int4.h"  // from @ml_dtypes
 
 namespace tsl {
 using float8_e4m3fn = ml_dtypes::float8_e4m3fn;
diff --git a/third_party/xla/xla/types.h b/third_party/xla/xla/types.h
index ef9dd11d..9cdc3965 100644
--- a/third_party/xla/xla/types.h
+++ b/third_party/xla/xla/types.h
@@ -23,7 +23,7 @@ limitations under the License.
 
 #include "absl/strings/str_format.h"
 #include "Eigen/Core"  // from @eigen_archive            // IWYU pragma: export
-#include "ml_dtypes/include/int4.h"  // from @ml_dtypes  // IWYU pragma: export
+#include "ml_dtypes/int4.h"  // from @ml_dtypes
 
 namespace xla {
 
