diff --git a/tensorflow/lite/CMakeLists.txt b/tensorflow/lite/CMakeLists.txt
index 09a3ce29..3611d7da 100644
--- a/tensorflow/lite/CMakeLists.txt
+++ b/tensorflow/lite/CMakeLists.txt
@@ -670,6 +670,39 @@ target_compile_options(tensorflow-lite
 add_library(tensorflow::tensorflowlite ALIAS tensorflow-lite)
 
 include(GNUInstallDirs)
+
+if(TFLITE_INSTALL_PRIVATE_HEADERS)
+    # Private headers. Currently, using WINDOWS_EXPORT_ALL_SYMBOLS for them.
+    install(DIRECTORY   ${TFLITE_SOURCE_DIR}/core
+            DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/tensorflow/lite
+            FILES_MATCHING PATTERN "*.h"
+    )
+    install(DIRECTORY   ${TFLITE_SOURCE_DIR}/kernels
+            DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/tensorflow/lite
+            FILES_MATCHING PATTERN "*.h"
+    )
+    install(DIRECTORY   ${TFLITE_SOURCE_DIR}/schema
+            DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/tensorflow/lite
+            FILES_MATCHING PATTERN "*.h"
+    )
+    install(DIRECTORY   ${TFLITE_SOURCE_DIR}/internal
+            DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/tensorflow/lite
+            FILES_MATCHING PATTERN "*.h"
+    )
+    install(DIRECTORY   ${TFLITE_SOURCE_DIR}/experimental/resource
+            DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/tensorflow/lite/experimental
+            FILES_MATCHING PATTERN "*.h"
+    )
+    install(DIRECTORY   ${TFLITE_SOURCE_DIR}/delegates/gpu
+            DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/tensorflow/lite/delegates
+            FILES_MATCHING PATTERN "*.h"
+    )
+    file(GLOB TFLITE_HEADERS "${TFLITE_SOURCE_DIR}/*.h")
+    install(FILES   ${TFLITE_HEADERS}
+            DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/tensorflow/lite
+    )
+endif()
+
 # C API
 install(DIRECTORY   ${TFLITE_SOURCE_DIR}/c
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/tensorflow/lite
