diff --git a/CMakeLists.txt b/CMakeLists.txt
index 96e0b0f..aec47e9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,7 @@
 # Licensed under the MIT License.
 cmake_minimum_required(VERSION 3.14)
 project(openclon12)
-include(FetchContent)
+include(GNUInstallDirs)
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
@@ -17,30 +17,10 @@ file(GLOB TRANS_SRC CONFIGURE_DEPENDS src/d3d12translationlayer/*.cpp)
 file(GLOB TRANS_INC include/d3d12translationlayer/*.*)
 file(GLOB TRANS_INL include/d3d12translationlayer/*.inl src/d3d12translationlayer/*.inl)
 
-include(FetchContent)
-FetchContent_Declare(
-    DirectX-Headers
-    GIT_REPOSITORY https://github.com/Microsoft/DirectX-Headers.git
-    GIT_TAG v1.610.0
-)
-FetchContent_MakeAvailable(DirectX-Headers)
-
-FetchContent_Declare(
-    opencl_headers
-    GIT_REPOSITORY https://github.com/KhronosGroup/OpenCL-Headers.git
-    GIT_TAG 1bb9ec797d14abed6167e3a3d66ede25a702a5c7
-)
-FetchContent_MakeAvailable(opencl_headers)
-add_library(OpenCL::Headers ALIAS Headers)
-
-set(WIL_BUILD_PACKAGING OFF CACHE BOOL "" FORCE)
-set(WIL_BUILD_TESTS OFF CACHE BOOL "" FORCE)
-FetchContent_Declare(
-    wil
-    GIT_REPOSITORY https://github.com/microsoft/wil.git
-    GIT_TAG ed429e64eb3b91848bf19c17e1431c1b0f2c6d2b
-)
-FetchContent_MakeAvailable(wil)
+find_path(D3D12TRANSLATIONLAYER_INCLUDE_DIR NAMES D3D12TranslationLayer/D3D12TranslationLayerIncludes.h REQUIRED)
+find_library(D3D12TRANSLATIONLAYER_LIB NAMES d3d12translationlayer REQUIRED)
+find_path(OpenCL_INCLUDE_DIR NAMES CL/cl.h REQUIRED)
+find_package(wil CONFIG REQUIRED) # WIL::WIL
 
 add_library(d3d12translationlayer STATIC ${TRANS_SRC} ${TRANS_INC} ${TRANS_INL})
 target_include_directories(d3d12translationlayer
@@ -56,15 +36,18 @@ add_library(openclon12 SHARED ${MAIN_SRC} ${MAIN_INC} ${EXTERNAL_INC})
 target_include_directories(openclon12
     PRIVATE src/openclon12
     PRIVATE external
-    PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
+    PRIVATE ${CMAKE_CURRENT_BINARY_DIR} ${D3D12TRANSLATIONLAYER_INCLUDE_DIR}/D3D12TranslationLayer ${OpenCL_INCLUDE_DIR})
 target_link_libraries(openclon12
-    d3d12translationlayer
-    OpenCL::Headers
-    WIL
+    PRIVATE ${D3D12TRANSLATIONLAYER_LIB} WIL::WIL dxcore d3d12 dxgi
     user32
     gdi32)
 source_group("Header Files\\External" FILES ${EXTERNAL_INC})
 
+install(FILES ${TRANS_INC} ${TRANS_INL} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/OpenCLOn12)
+install(TARGETS openclon12
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
 option(BUILD_TESTS "Build tests" ON)
 
 if (BUILD_TESTS)
