diff --git a/CMakeLists.txt b/CMakeLists.txt
index 114d793..7a05453 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -668,7 +668,7 @@ include(CTest)
 enable_testing()
 
 if (CUTLASS_ENABLE_GTEST_UNIT_TESTS)
-  include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/googletest.cmake)
+  find_package(GTest CONFIG REQUIRED) # GTest::gtest GTest::gmock
 endif()
 
 if (NOT TARGET test_all)
diff --git a/test/unit/CMakeLists.txt b/test/unit/CMakeLists.txt
index 16252a0..800ffc6 100644
--- a/test/unit/CMakeLists.txt
+++ b/test/unit/CMakeLists.txt
@@ -42,7 +42,7 @@ target_link_libraries(
   CUTLASS
   cutlass_tools_util_includes
   $<$<BOOL:${CUTLASS_ENABLE_CUBLAS}>:nvidia::cublas>
-  gtest
+  GTest::gtest
   cudart
   cuda_driver
   )
@@ -84,7 +84,7 @@ function(cutlass_test_unit_add_executable NAME)
     target_link_libraries(
       ${NAME}
       PUBLIC
-      gtest
+      GTest::gtest
     )
   else()
     target_link_libraries(
diff --git a/test/unit/core/CMakeLists.txt b/test/unit/core/CMakeLists.txt
index 965c050..a11c0e8 100644
--- a/test/unit/core/CMakeLists.txt
+++ b/test/unit/core/CMakeLists.txt
@@ -45,6 +45,10 @@ cutlass_test_unit_add_executable(
   functional.cu
   )
 
+target_include_directories(cutlass_test_unit_core PUBLIC
+  $<BUILD_INTERFACE:${CUTLASS_INCLUDE_DIR}>
+  $<BUILD_INTERFACE:${CUTLASS_TOOLS_UTIL_INCLUDE_DIR}>
+)
 #
 # CUTLASS 3x increases the host compiler requirements to C++17. However, there are
 # certain existing integrations that will benefit from maintaining C++11 compatibility.
