diff --git a/mediapipe/gpu/gl_calculator_helper.cc b/mediapipe/gpu/gl_calculator_helper.cc
index efd822a..f8ac8aa 100644
--- a/mediapipe/gpu/gl_calculator_helper.cc
+++ b/mediapipe/gpu/gl_calculator_helper.cc
@@ -24,6 +24,10 @@
 #include "mediapipe/gpu/gpu_buffer.h"
 #include "mediapipe/gpu/gpu_service.h"
 
+#if defined(GetObject)
+#undef GetObject
+#endif
+
 namespace mediapipe {
 
 // The constructor and destructor need to be defined here so that
