diff --git a/mediapipe/gpu/gl_calculator_helper.cc b/mediapipe/gpu/gl_calculator_helper.cc
index efd822a..f8ac8aa 100644
--- a/mediapipe/gpu/gl_calculator_helper.cc
+++ b/mediapipe/gpu/gl_calculator_helper.cc
@@ -24,6 +24,10 @@
 #include "mediapipe/gpu/gpu_buffer.h"
 #include "mediapipe/gpu/gpu_service.h"
 
+// https://docs.microsoft.com/en-us/windows/win32/api/wingdi/nf-wingdi-getobjecta
+#if defined(_WIN32)
+#undef GetObject
+#endif
 namespace mediapipe {
 
 // The constructor and destructor need to be defined here so that
diff --git a/mediapipe/gpu/gl_context.cc b/mediapipe/gpu/gl_context.cc
index c27a8b4..1f0702b 100644
--- a/mediapipe/gpu/gl_context.cc
+++ b/mediapipe/gpu/gl_context.cc
@@ -447,7 +447,8 @@ void GlContext::RunWithoutWaiting(GlVoidFunction gl_func) {
 
 std::weak_ptr<GlContext>& GlContext::CurrentContext() {
   // Workaround for b/67878799.
-#ifndef __EMSCRIPTEN__
+  // Windows build doesn't export the symbols
+#if !defined(_WIN32)
   absl::LeakCheckDisabler disable_leak_check;
 #endif
   ABSL_CONST_INIT thread_local std::weak_ptr<GlContext> current_context;
