diff --git a/cmake/onnxruntime_providers_tensorrt.cmake b/cmake/onnxruntime_providers_tensorrt.cmake
index 59c7db9..44d14fd 100644
--- a/cmake/onnxruntime_providers_tensorrt.cmake
+++ b/cmake/onnxruntime_providers_tensorrt.cmake
@@ -28,7 +28,7 @@
   endif()
   set(CXX_VERSION_DEFINED TRUE)
 
-  find_path(TENSORRT_INCLUDE_DIR NvInfer.h
+  find_path(TENSORRT_INCLUDE_DIR NvInfer.h REQUIRED
     HINTS ${TENSORRT_ROOT}
     PATH_SUFFIXES include)
 
