diff --git a/cmake/tensorboard/compat/proto/CMakeLists.txt b/cmake/tensorboard/compat/proto/CMakeLists.txt
index ad31e40..ece628b 100644
--- a/cmake/tensorboard/compat/proto/CMakeLists.txt
+++ b/cmake/tensorboard/compat/proto/CMakeLists.txt
@@ -16,7 +16,7 @@ file(GLOB_RECURSE tensorboard_proto_srcs CONFIGURE_DEPENDS
 )
 
 add_library(tensorboard STATIC ${tensorboard_proto_srcs})
-onnxruntime_protobuf_generate(APPEND_PATH IMPORT_DIRS ${tensorboard_SOURCE_DIR} ${protobuf_SOURCE_DIR}/src TARGET tensorboard)
+onnxruntime_protobuf_generate(APPEND_PATH IMPORT_DIRS ${tensorboard_SOURCE_DIR} TARGET tensorboard)
 onnxruntime_add_include_to_target(tensorboard ${PROTOBUF_LIB})
 target_include_directories(tensorboard PRIVATE ${PROJECT_BINARY_DIR})
 add_dependencies(tensorboard ${onnxruntime_EXTERNAL_DEPENDENCIES})
