diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 4b86657..4a53ecf 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -2031,9 +2031,9 @@ endif()
 if (WIN32 AND NOT GDK_PLATFORM)
   add_compile_definitions(WINAPI_FAMILY=100) # Desktop app
   if (onnxruntime_USE_WINML)
-    add_compile_definitions(WINVER=0x0602 _WIN32_WINNT=0x0602 NTDDI_VERSION=0x06020000)  # Support Windows 8 and newer
+    add_compile_definitions(WINVER=0x0A00 _WIN32_WINNT=0x0A00 NTDDI_VERSION=0x06020000)  # Support Windows 10 and newer
   else()
-    add_compile_definitions(WINVER=0x0601 _WIN32_WINNT=0x0601 NTDDI_VERSION=0x06010000)  # Support Windows 7 and newer
+    add_compile_definitions(WINVER=0x0A00 _WIN32_WINNT=0x0A00 NTDDI_VERSION=0x06010000)  # Support Windows 10 and newer
   endif()
   if (NOT CMAKE_CXX_STANDARD_LIBRARIES MATCHES kernel32.lib)
     # On onecore, link to the onecore build of the MSVC runtime
