diff --git a/CMakeLists.txt b/CMakeLists.txt
index abc123..def456 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -208,8 +208,8 @@ cpp_library(
   DEFINITIONS
     ${fbgemm_generic_defs}
   DEPS
-    $<BUILD_INTERFACE:cpuinfo>
-    $<BUILD_INTERFACE:asmjit>
+    cpuinfo::cpuinfo
+    asmjit::asmjit
 )
 
 set(fbgemm_targets)
@@ -238,8 +238,8 @@ cpp_library(
     MSVC_FLAGS
       ${CXX_AVX2_FLAGS}
     DEPS
-      $<BUILD_INTERFACE:cpuinfo>
-      $<BUILD_INTERFACE:asmjit>
+      cpuinfo::cpuinfo
+      asmjit::asmjit
       fbgemm_generic
   )
 
@@ -272,8 +272,8 @@ cpp_library(
     MSVC_FLAGS
       ${CXX_AVX512_FLAGS} ${CXX_AVX2_FLAGS}
     DEPS
-      $<BUILD_INTERFACE:cpuinfo>
-      $<BUILD_INTERFACE:asmjit>
+      cpuinfo::cpuinfo
+      asmjit::asmjit
       fbgemm_generic
   )
 
@@ -321,8 +321,8 @@ cpp_library(
     DEFINITIONS
       ${fbgemm_arm_defs}
     DEPS
-      $<BUILD_INTERFACE:cpuinfo>
-      $<BUILD_INTERFACE:asmjit>
+      cpuinfo::cpuinfo
+      asmjit::asmjit
       fbgemm_generic
   )
 
@@ -378,8 +378,8 @@ cpp_library(
     DEFINITIONS
       ${fbgemm_arm_defs}
     DEPS
-      $<BUILD_INTERFACE:cpuinfo>
-      $<BUILD_INTERFACE:asmjit>
+      cpuinfo::cpuinfo
+      asmjit::asmjit
       fbgemm_generic
   )
 
@@ -410,8 +410,8 @@ cpp_library(
     ENABLE_IPO
       ${FBGEMM_USE_IPO}
     DEPS
-      $<BUILD_INTERFACE:cpuinfo>
-      $<BUILD_INTERFACE:asmjit>
+      cpuinfo::cpuinfo
+      asmjit::asmjit
       fbgemm_generic
   )
 
@@ -447,8 +447,8 @@ install(
 # Asmjit Target
 ################################################################################
 
-if(NOT TARGET asmjit)
-  # Download asmjit from github if ASMJIT_SRC_DIR is not specified.
+find_package(asmjit CONFIG REQUIRED)
+if(FALSE)
   if(NOT DEFINED ASMJIT_SRC_DIR)
     set(ASMJIT_SRC_DIR "${FBGEMM_SOURCE_DIR}/external/asmjit"
       CACHE STRING "asmjit source directory from submodules")
@@ -469,8 +469,8 @@ endif()
 # Cpuinfo Target
 ################################################################################
 
-if(NOT TARGET cpuinfo)
-  #Download cpuinfo from github if CPUINFO_SOURCE_DIR is not specified.
+find_package(cpuinfo CONFIG REQUIRED)
+if(FALSE)
   if(NOT DEFINED CPUINFO_SOURCE_DIR)
     set(CPUINFO_SOURCE_DIR "${FBGEMM_SOURCE_DIR}/external/cpuinfo"
       CACHE STRING "cpuinfo source directory from submodules")
