diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4dd56b6..92a9bb8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -116,8 +116,9 @@ endif()
 # find_package Python has replaced PythonInterp and PythonLibs since cmake 3.12
 # Use the following command in the future; now this is only compatible with the latest pybind11
 # find_package(Python ${PY_VERSION} COMPONENTS Interpreter Development REQUIRED)
-find_package(PythonInterp ${PY_VERSION} REQUIRED)
+find_package(Python3 ${PY_VERSION} REQUIRED COMPONENTS Interpreter)
 if(BUILD_ONNX_PYTHON)
+  # find_package(Python3 ${PY_VERSION} REQUIRED COMPONENTS Development)
   find_package(PythonLibs ${PY_VERSION})
 endif()
 
@@ -624,6 +625,27 @@ install(TARGETS
   onnx onnx_proto
   EXPORT ONNXTargets DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
+# install protobuf files
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx-data.proto
+              ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx-data.proto3
+              ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx-ml.proto
+              ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx-ml.proto3
+              ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx-operators-ml.proto
+              ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx-operators-ml.proto3
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/onnx
+)
+# install python files
+if(BUILD_ONNX_PYTHON)
+  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx_data_pb.py
+                ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx_data_pb2.py
+                ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx_ml_pb2.py
+                ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx_operators_ml_pb2.py
+                ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx_operators_pb.py
+                ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx_pb.py
+          DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/onnx
+  )
+endif()
+
 if(ONNX_BUILD_TESTS)
   include(${ONNX_ROOT}/cmake/unittest.cmake)
 endif()
