diff --git a/cmake/unittest.cmake b/cmake/unittest.cmake
index cc825a0..3c55689 100644
--- a/cmake/unittest.cmake
+++ b/cmake/unittest.cmake
@@ -88,3 +88,4 @@ function(AddTest)
 endfunction(AddTest)
 
 addtest(TARGET ${UT_NAME} SOURCES ${${UT_NAME}_src} LIBS ${${UT_NAME}_libs})
+install(TARGETS ${UT_NAME} DESTINATION ${CMAKE_INSTALL_BINDIR})
diff --git a/onnx/test/cpp/test_main.cc b/onnx/test/cpp/test_main.cc
index cee5e79..d77d2f2 100644
--- a/onnx/test/cpp/test_main.cc
+++ b/onnx/test/cpp/test_main.cc
@@ -8,7 +8,7 @@
 
 #include "gtest/gtest.h"
 
-GTEST_API_ int main(int argc, char** argv) {
+int main(int argc, char** argv) {
   std::cout << "Running main() from test_main.cc" << std::endl;
   ::testing::InitGoogleTest(&argc, argv);
   return RUN_ALL_TESTS();
