diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0063385..e9ad9ff 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,7 +26,7 @@ if(NOT ONNX_OPT_USE_SYSTEM_PROTOBUF)
   add_subdirectory_if_no_target(${PROJECT_SOURCE_DIR}/third_party/protobuf/cmake libprotobuf)
 endif()
 
-
+find_package(ONNX CONFIG REQUIRED) # ONNX::onnx
 set(ONNX_ROOT ${PROJECT_SOURCE_DIR}/third_party/onnx)
 add_subdirectory_if_no_target(${ONNX_ROOT} ${ONNX_TARGET_NAME})
 
diff --git a/cmake/utils.cmake b/cmake/utils.cmake
index 6cca9f3..a735434 100644
--- a/cmake/utils.cmake
+++ b/cmake/utils.cmake
@@ -1,4 +1,10 @@
-include(${PROJECT_SOURCE_DIR}/third_party/onnx/cmake/Utils.cmake)
+function(add_msvc_runtime_flag lib)
+  if(ONNX_USE_MSVC_STATIC_RUNTIME)
+    target_compile_options(${lib} PRIVATE $<$<NOT:$<CONFIG:Debug>>:/MT> $<$<CONFIG:Debug>:/MTd>)    
+  else()
+    target_compile_options(${lib} PRIVATE $<$<NOT:$<CONFIG:Debug>>:/MD> $<$<CONFIG:Debug>:/MDd>)
+  endif()
+endfunction()
 
 # Poor man's FetchContent
 function(add_subdirectory_if_no_target dir target)
