diff -Naur optimizer-0.4.0-orig/CMakeLists.txt optimizer-0.4.0/CMakeLists.txt
--- optimizer-0.4.0-orig/CMakeLists.txt	2026-01-03 13:47:14.271236880 +0000
+++ optimizer-0.4.0/CMakeLists.txt	2026-01-03 13:47:58.141412656 +0000
@@ -12,6 +12,10 @@
 
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
+function(add_msvc_runtime_flag lib)
+    # Dummy function to mock embedded ONNX project build. Let CMake decide /MT, /MD, /MTd, /MDd, etc.
+endfunction()
+
 include(cmake/utils.cmake)
 
 # For integration with onnxruntime_webassembly etc.
@@ -25,14 +29,12 @@
 
 option(ONNX_OPT_USE_SYSTEM_PROTOBUF "" OFF)
 if(NOT ONNX_OPT_USE_SYSTEM_PROTOBUF)
-  option(protobuf_BUILD_TESTS "" OFF)
-  option(protobuf_MSVC_STATIC_RUNTIME "" ${ONNX_USE_MSVC_STATIC_RUNTIME})
-  add_subdirectory_if_no_target(${PROJECT_SOURCE_DIR}/third_party/protobuf libprotobuf)
 endif()
 
 
-set(ONNX_ROOT ${PROJECT_SOURCE_DIR}/third_party/onnx)
-add_subdirectory_if_no_target(${ONNX_ROOT} ${ONNX_TARGET_NAME})
+
+find_package(protobuf CONFIG REQUIRED)
+find_package(ONNX CONFIG REQUIRED)
 
 file(READ "${PROJECT_SOURCE_DIR}/VERSION_NUMBER" ONNX_OPTIMIZER_VERSION)
 string(STRIP "${ONNX_OPTIMIZER_VERSION}" ONNX_OPTIMIZER_VERSION)
@@ -45,7 +47,7 @@
 list(REMOVE_ITEM onnx_opt_srcs "${PROJECT_SOURCE_DIR}/onnxoptimizer/cpp2py_export.cc")
 
 onnxopt_add_library(onnx_optimizer ${onnx_opt_srcs})
-target_link_libraries(onnx_optimizer PUBLIC ${ONNX_TARGET_NAME})
+target_link_libraries(onnx_optimizer PUBLIC ONNX::onnx ONNX::onnx_proto)
 target_include_directories(onnx_optimizer PUBLIC
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
     $<INSTALL_INTERFACE:include>
diff -Naur optimizer-0.4.0-orig/cmake/ONNXOptimizerConfig.cmake.in optimizer-0.4.0/cmake/ONNXOptimizerConfig.cmake.in
--- optimizer-0.4.0-orig/cmake/ONNXOptimizerConfig.cmake.in	2026-01-03 13:47:14.271236880 +0000
+++ optimizer-0.4.0/cmake/ONNXOptimizerConfig.cmake.in	2026-01-03 13:48:19.264495804 +0000
@@ -3,6 +3,10 @@
 #   ONNX_OPTIMIZER_INCLUDE_DIRS     - include directories for onnx optimizer
 # as well as ONNX Optimizer targets for other cmake libraries to use.
 
+include(CMakeFindDependencyMacro)
+find_dependency(protobuf CONFIG)
+find_dependency(ONNX CONFIG)
+
 # library version information
 set(ONNX_OPTIMIZER_VERSION "@ONNX_OPTIMIZER_VERSION@")
 
diff -Naur optimizer-0.4.0-orig/cmake/utils.cmake optimizer-0.4.0/cmake/utils.cmake
--- optimizer-0.4.0-orig/cmake/utils.cmake	2026-01-03 13:47:14.271236880 +0000
+++ optimizer-0.4.0/cmake/utils.cmake	2026-01-03 13:48:07.815451402 +0000
@@ -2,7 +2,6 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-include(${PROJECT_SOURCE_DIR}/third_party/onnx/cmake/Utils.cmake)
 
 # Poor man's FetchContent
 function(add_subdirectory_if_no_target dir target)
