diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1234567..abcdefg 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -390,7 +390,7 @@ ENDIF()
 # ---[ Build flags
 IF(NOT CMAKE_SYSTEM_NAME)
   MESSAGE(FATAL_ERROR "CMAKE_SYSTEM_NAME not defined")
-ELSEIF(NOT CMAKE_SYSTEM_NAME MATCHES "^(Android|Darwin|iOS|Linux|Windows|CYGWIN|MSYS|QURT|Emscripten)$")
+ELSEIF(NOT CMAKE_SYSTEM_NAME MATCHES "^(Android|Darwin|iOS|Linux|Windows|WindowsStore|CYGWIN|MSYS|QURT|Emscripten)$")
   MESSAGE(FATAL_ERROR "Unrecognized CMAKE_SYSTEM_NAME value \"${CMAKE_SYSTEM_NAME}\"")
 ENDIF()
 IF(CMAKE_SYSTEM_NAME MATCHES "Windows")
@@ -1211,7 +1211,7 @@ ENDIF()
 INCLUDE_DIRECTORIES(.)
 TARGET_INCLUDE_DIRECTORIES(xnnpack-hardware-config PRIVATE include src ${CPUINFO_SOURCE_DIR}/include)
 IF(XNNPACK_BUILD_LIBRARY)
-  TARGET_INCLUDE_DIRECTORIES(XNNPACK PUBLIC include)
+  target_include_directories(XNNPACK PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>)
   IF(WIN32)
     # Target Windows 7+ API
     TARGET_COMPILE_DEFINITIONS(XNNPACK PRIVATE _WIN32_WINNT=0x0601)
@@ -1290,7 +1290,11 @@ TARGET_LINK_LIBRARIES(xnnpack-base INTERFACE pthreadpool)
 
 # ---[ Configure FXdiv
 IF(NOT TARGET fxdiv)
-  IF(NOT XNNPACK_USE_SYSTEM_LIBS)
+  if(TRUE)
+    find_path(FXDIV_INCLUDE_DIR fxdiv.h REQUIRED)
+    add_library(fxdiv INTERFACE)
+    target_include_directories(fxdiv INTERFACE "${FXDIV_INCLUDE_DIR}")
+  elseif(NOT XNNPACK_USE_SYSTEM_LIBS)
     SET(FXDIV_BUILD_TESTS OFF CACHE BOOL "")
     SET(FXDIV_BUILD_BENCHMARKS OFF CACHE BOOL "")
     ADD_SUBDIRECTORY(
  diff --git a/tools/update-microkernels.py b/tools/update-microkernels.py
  index 1111111..2222222 100644
  --- a/tools/update-microkernels.py
  +++ b/tools/update-microkernels.py
  @@ -117,10 +117,10 @@ def overwrite_if_changed(filepath, content):
    txt_changed = True
    if os.path.exists(filepath):
  -    with codecs.open(filepath, 'r', encoding='utf-8') as output_file:
  +    with open(filepath, 'r', encoding='utf-8') as output_file:
       txt_changed = output_file.read() != content
    if txt_changed:
  -    with codecs.open(filepath, 'w', encoding='utf-8') as output_file:
  +    with open(filepath, 'w', encoding='utf-8') as output_file:
       output_file.write(content)
