diff --git a/CMakeLists.txt b/CMakeLists.txt
index 69210a2..702af97 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7003,6 +7003,15 @@ INSTALL(TARGETS XNNPACK
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
+# support find_package(xnnpack CONFIG)
+include(CMakePackageConfigHelpers)
+get_filename_component(CONFIG_FILE_PATH ${CMAKE_CURRENT_BINARY_DIR}/xnnpack-config.cmake ABSOLUTE)
+configure_package_config_file(
+  cmake/xnnpack-config.cmake.in ${CONFIG_FILE_PATH}
+  INSTALL_DESTINATION ${CMAKE_INSTALL_DATAROOTDIR})
+install(FILES ${CONFIG_FILE_PATH}
+  DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}) # ${prefix}/share
+
 # ---[ XNNPACK unit tests
 IF(XNNPACK_BUILD_TESTS)
   # ---[ Build google test
