diff --git a/CMakeLists.txt b/CMakeLists.txt
index d4b0602..140d105 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,7 +61,7 @@ ADD_COMPILE_DEFINITIONS("XNN_ENABLE_GEMM_M_SPECIALIZATION=$<BOOL:${XNNPACK_ENABL
 
 IF(CMAKE_C_COMPILER_ID STREQUAL "MSVC")
   # Disable "unary minus operator applied to unsigned type, result still unsigned" warning
-  ADD_COMPILE_OPTIONS("/wd4146")
+  ADD_COMPILE_OPTIONS("/wd4146" "/MP" "/Qpar")
 ENDIF()
 
 IF(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
