diff --git a/src/c++/CMakeLists.txt b/src/c++/CMakeLists.txt
index a542531..7fc7041 100644
--- a/src/c++/CMakeLists.txt
+++ b/src/c++/CMakeLists.txt
@@ -57,19 +57,7 @@ endif()
 #
 # Dependencies
 #
-include(FetchContent)
-
-FetchContent_Declare(
-  repo-common
-  GIT_REPOSITORY ${TRITON_REPO_ORGANIZATION}/common.git
-  GIT_TAG ${TRITON_COMMON_REPO_TAG}
-  GIT_SHALLOW ON
-)
-
-FetchContent_Declare(
-  googletest
-  URL https://github.com/google/googletest/archive/9406a60c7839052e4944ea4dbc8344762a89f9bd.zip
-)
+find_package(TritonCommon CONFIG REQUIRED)
 
 if(TRITON_ENABLE_CC_GRPC OR TRITON_ENABLE_PERF_ANALYZER)
   set(TRITON_COMMON_ENABLE_PROTOBUF ON)
@@ -93,7 +81,6 @@ endif()
 if(TRITON_ENABLE_TESTS OR TRITON_ENABLE_PERF_ANALYZER)
   FetchContent_MakeAvailable(googletest)
 endif()
-FetchContent_MakeAvailable(repo-common)
 
 if(TRITON_ENABLE_TESTS)
   include_directories(
diff --git a/src/c++/library/CMakeLists.txt b/src/c++/library/CMakeLists.txt
index cdee03e..6033ec9 100644
--- a/src/c++/library/CMakeLists.txt
+++ b/src/c++/library/CMakeLists.txt
@@ -48,7 +48,7 @@ target_include_directories(
 if(TRITON_ENABLE_CC_HTTP OR TRITON_ENABLE_PERF_ANALYZER OR TRITON_ENABLE_EXAMPLES)
   find_package(RapidJSON CONFIG REQUIRED)
   add_library(
-      json-utils-library EXCLUDE_FROM_ALL OBJECT
+      json-utils-library OBJECT
       json_utils.h json_utils.cc
   )
   target_include_directories(
@@ -93,20 +93,14 @@ if(TRITON_ENABLE_CC_GRPC OR TRITON_ENABLE_PERF_ANALYZER)
   )
 
   add_library(
-      grpc-client-library EXCLUDE_FROM_ALL OBJECT
+      grpc-client-library OBJECT
       ${REQUEST_SRCS} ${REQUEST_HDRS}
   )
-  add_dependencies(
-      grpc-client-library
-      grpc-service-library proto-library
-  )
 
   # libgrpcclient_static.a
   add_library(
       grpcclient_static STATIC
       $<TARGET_OBJECTS:grpc-client-library>
-      $<TARGET_OBJECTS:grpc-service-library>
-      $<TARGET_OBJECTS:proto-library>
   )
   add_library(
       TritonClient::grpcclient_static ALIAS grpcclient_static
@@ -122,6 +116,8 @@ if(TRITON_ENABLE_CC_GRPC OR TRITON_ENABLE_PERF_ANALYZER)
       grpcclient_static
       PRIVATE gRPC::grpc++
       PRIVATE gRPC::grpc
+      PRIVATE TritonCommon::grpc-service-library
+      PRIVATE TritonCommon::proto-library
       PUBLIC protobuf::libprotobuf
       PUBLIC Threads::Threads
   )
@@ -129,8 +125,6 @@ if(TRITON_ENABLE_CC_GRPC OR TRITON_ENABLE_PERF_ANALYZER)
   # libgrpcclient.so
   add_library(
       grpcclient SHARED
-      $<TARGET_OBJECTS:grpc-service-library>
-      $<TARGET_OBJECTS:proto-library>
       $<TARGET_OBJECTS:grpc-client-library>
   )
   add_library(
@@ -150,6 +144,8 @@ if(TRITON_ENABLE_CC_GRPC OR TRITON_ENABLE_PERF_ANALYZER)
       grpcclient
       PRIVATE gRPC::grpc++
       PRIVATE gRPC::grpc
+      PRIVATE TritonCommon::grpc-service-library
+      PRIVATE TritonCommon::proto-library
       PUBLIC protobuf::libprotobuf
       PUBLIC Threads::Threads
   )
@@ -169,13 +165,13 @@ if(TRITON_ENABLE_CC_GRPC OR TRITON_ENABLE_PERF_ANALYZER)
         POSITION_INDEPENDENT_CODE ON
     )
 
+    find_path(TRITON_GRPC_INCLUDE_DIR NAMES grpc_service.grpc.pb.h REQUIRED)
     target_include_directories(
       ${_client_target}
       PUBLIC
         $<INSTALL_INTERFACE:include>
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-        $<TARGET_PROPERTY:proto-library,INCLUDE_DIRECTORIES>
-        $<TARGET_PROPERTY:grpc-service-library,INCLUDE_DIRECTORIES>
+        ${TRITON_GRPC_INCLUDE_DIR}
       PRIVATE
         ${CMAKE_CURRENT_SOURCE_DIR}
     )
@@ -234,20 +230,12 @@ if(TRITON_ENABLE_CC_GRPC OR TRITON_ENABLE_PERF_ANALYZER)
       )
     endif() # NOT WIN32
 
-    install(
-      DIRECTORY
-        ${CMAKE_CURRENT_BINARY_DIR}/../../third-party/grpc/include/
-        ${CMAKE_CURRENT_BINARY_DIR}/../../third-party/absl/include/
-        ${CMAKE_CURRENT_BINARY_DIR}/../../third-party/protobuf/include/
-        ${CMAKE_CURRENT_BINARY_DIR}/../../third-party/re2/include/
-        DESTINATION include
-    )
   endif()
 
   install(
       FILES
       ${CMAKE_CURRENT_SOURCE_DIR}/grpc_client.h
-      DESTINATION include
+      DESTINATION include/triton
   )
 endif() # TRITON_ENABLE_CC_GRPC OR TRITON_ENABLE_PERF_ANALYZER
 
@@ -272,7 +260,7 @@ if(TRITON_ENABLE_CC_HTTP OR TRITON_ENABLE_PERF_ANALYZER)
   )
 
   add_library(
-      http-client-library EXCLUDE_FROM_ALL OBJECT
+      http-client-library STATIC
       ${REQUEST_SRCS} ${REQUEST_HDRS}
   )
 
@@ -286,7 +274,7 @@ if(TRITON_ENABLE_CC_HTTP OR TRITON_ENABLE_PERF_ANALYZER)
   target_link_libraries(
     http-client-library
     PUBLIC
-      triton-common-json        # from repo-common
+      TritonCommon::triton-common-json # from repo-common
   )
 
   # libhttpclient_static.a
@@ -300,7 +288,7 @@ if(TRITON_ENABLE_CC_HTTP OR TRITON_ENABLE_PERF_ANALYZER)
 
   target_link_libraries(
       httpclient_static
-      PRIVATE triton-common-json
+      PRIVATE TritonCommon::triton-common-json
       PUBLIC CURL::libcurl
       PUBLIC Threads::Threads
   )
@@ -332,7 +320,7 @@ if(TRITON_ENABLE_CC_HTTP OR TRITON_ENABLE_PERF_ANALYZER)
 
   target_link_libraries(
       httpclient
-      PRIVATE triton-common-json
+      PRIVATE TritonCommon::triton-common-json
       PUBLIC CURL::libcurl
       PUBLIC Threads::Threads
   )
@@ -391,7 +379,7 @@ if(TRITON_ENABLE_CC_HTTP OR TRITON_ENABLE_PERF_ANALYZER)
   install(
       FILES
       ${CMAKE_CURRENT_SOURCE_DIR}/http_client.h
-      DESTINATION include
+      DESTINATION include/triton
   )
 endif() # TRITON_ENABLE_CC_HTTP OR TRITON_ENABLE_PERF_ANALYZER
 
@@ -400,7 +388,7 @@ if(TRITON_ENABLE_CC_HTTP OR TRITON_ENABLE_CC_GRPC OR TRITON_ENABLE_PERF_ANALYZER
       FILES
       ${CMAKE_CURRENT_SOURCE_DIR}/common.h
       ${CMAKE_CURRENT_SOURCE_DIR}/ipc.h
-      DESTINATION include
+      DESTINATION include/triton
   )
 
   include(GNUInstallDirs)
