diff --git a/mlmodel/tools/enumgen.cpp b/mlmodel/tools/enumgen.cpp
index f6feb4b..5c07c55 100644
--- a/mlmodel/tools/enumgen.cpp
+++ b/mlmodel/tools/enumgen.cpp
@@ -1,4 +1,4 @@
-#include "../../deps/protobuf/src/google/protobuf/compiler/plugin.pb.h"
+#include <google/protobuf/compiler/plugin.pb.h>
 
 #include <iostream>
 #include <sstream>
diff --git a/modelpackage/src/utils/JsonMap.cpp b/modelpackage/src/utils/JsonMap.cpp
index 400dcf2..c28e61c 100644
--- a/modelpackage/src/utils/JsonMap.cpp
+++ b/modelpackage/src/utils/JsonMap.cpp
@@ -11,7 +11,7 @@
 #include <sstream>
 
 #include "JsonMap.hpp"
-#include "json.hpp"
+#include <nlohmann/json.hpp>
 
 using namespace nlohmann;
 
