diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8f58e79..0ed0aad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Boilerplate: Copyright (C) 2014-2017 Ruslan Baratov
 # Modification: Copyright (C) 2017 David Hirvonen
 
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.22)
 project(NEON_2_SSE VERSION 1.0.0 LANGUAGES C)
 
 add_library(${PROJECT_NAME} INTERFACE)
