diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index d64956b..c412c84 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -1,6 +1,14 @@
 # dependencies
 
 find_package(Threads REQUIRED)
+if(LLAMA_SERVER_SSL)
+    find_package(ZLIB REQUIRED)
+    find_package(OpenSSL REQUIRED)
+    link_libraries(ZLIB::ZLIB OpenSSL::SSL OpenSSL::Crypto)
+    if(APPLE)
+        link_libraries("-framework Security" "-framework SystemConfiguration")
+    endif()
+endif()
 
 # third-party
 
