diff --git a/scripts/build-info.cmake b/scripts/build-info.cmake
index c86ab43..0571c20 100644
--- a/scripts/build-info.cmake
+++ b/scripts/build-info.cmake
@@ -1,9 +1,5 @@
 set(TEMPLATE_FILE "${CMAKE_CURRENT_SOURCE_DIR}/scripts/build-info.h.in")
 set(HEADER_FILE "${CMAKE_CURRENT_SOURCE_DIR}/build-info.h")
-set(BUILD_NUMBER 0)
-set(BUILD_COMMIT "unknown")
-set(BUILD_COMPILER "unknown")
-set(BUILD_TARGET "unknown")
 
 # Look for git
 find_package(Git)
@@ -18,22 +14,7 @@ if(NOT Git_FOUND)
 endif()
 
 # Get the commit count and hash
-if(Git_FOUND)
-    execute_process(
-        COMMAND ${GIT_EXECUTABLE} rev-parse --short HEAD
-        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-        OUTPUT_VARIABLE HEAD
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-    )
-    execute_process(
-        COMMAND ${GIT_EXECUTABLE} rev-list --count HEAD
-        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-        OUTPUT_VARIABLE COUNT
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-    )
-    set(BUILD_COMMIT ${HEAD})
-    set(BUILD_NUMBER ${COUNT})
-endif()
+# see portfile.cmake
 
 if(MSVC)
     set(BUILD_COMPILER "${CMAKE_C_COMPILER_ID} ${CMAKE_C_COMPILER_VERSION}")
@@ -72,3 +53,16 @@ if(EXISTS ${HEADER_FILE})
 else()
     configure_file(${TEMPLATE_FILE} ${HEADER_FILE})
 endif()
+
+if(NOT DEFINED BUILD_NUMBER)
+    set(BUILD_NUMBER 0)
+endif()
+if(NOT DEFINED BUILD_COMMIT)
+    set(BUILD_COMMIT "unknown")
+endif()
+if(NOT DEFINED BUILD_COMPILER)
+    set(BUILD_COMPILER "unknown")
+endif()
+if(NOT DEFINED BUILD_TARGET)
+    set(BUILD_TARGET "unknown")
+endif()
