diff --git a/CMakeLists.txt b/CMakeLists.txt
index edd0ea1..3aadeec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -201,7 +201,7 @@ add_subdirectory(src)
 
 if (LLAMA_BUILD_COMMON)
     add_subdirectory(common)
-    add_subdirectory(vendor/cpp-httplib)
+    find_package(httplib CONFIG REQUIRED GLOBAL)
 endif()
 
 if (LLAMA_BUILD_COMMON AND LLAMA_BUILD_TESTS AND NOT CMAKE_JS_VERSION)
diff --git a/common/CMakeLists.txt b/common/CMakeLists.txt
index 1a56c25..d28413a 100644
--- a/common/CMakeLists.txt
+++ b/common/CMakeLists.txt
@@ -124,7 +124,9 @@ set_target_properties(${TARGET} PROPERTIES
     MACHO_CURRENT_VERSION 0 # keep macOS linker from seeing oversized version number
 )
 
-target_include_directories(${TARGET} PUBLIC . ../vendor)
+target_include_directories(${TARGET} PUBLIC "${PROJECT_SOURCE_DIR}/vendor")
+find_path(MINJA_INCLUDE_DIRS "minja/chat-template.hpp" REQUIRED)
+target_include_directories(${TARGET} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR} ${MINJA_INCLUDE_DIRS})
 target_compile_features   (${TARGET} PUBLIC cxx_std_17)
 
 if (BUILD_SHARED_LIBS)
@@ -135,7 +137,8 @@ if (BUILD_SHARED_LIBS)
 endif()
 
 target_link_libraries(${TARGET} PUBLIC  llama-common-base)
-target_link_libraries(${TARGET} PRIVATE cpp-httplib)
+find_package(httplib CONFIG REQUIRED GLOBAL)
+target_link_libraries(${TARGET} PRIVATE httplib::httplib)
 
 if (LLAMA_LLGUIDANCE)
     include(ExternalProject)
diff --git a/common/http.h b/common/http.h
index d3daccd..be18264 100644
--- a/common/http.h
+++ b/common/http.h
@@ -1,6 +1,6 @@
 #pragma once
 
-#include <cpp-httplib/httplib.h>
+#include <httplib.h>
 
 struct common_http_url {
     std::string scheme;
diff --git a/tools/mtmd/CMakeLists.txt b/tools/mtmd/CMakeLists.txt
index ffd30c7..72641cb 100644
--- a/tools/mtmd/CMakeLists.txt
+++ b/tools/mtmd/CMakeLists.txt
@@ -55,7 +55,9 @@ target_link_libraries     (mtmd PUBLIC ggml llama)
 target_link_libraries     (mtmd PRIVATE Threads::Threads)
 target_include_directories(mtmd PUBLIC  .)
 target_include_directories(mtmd PRIVATE ../..)
-target_include_directories(mtmd PRIVATE ../../vendor)
+find_path(STB_INCLUDE_DIR stb_image.h REQUIRED)
+find_path(MINIAUDIO_INCLUDE_DIR miniaudio.h REQUIRED)
+target_include_directories(mtmd PUBLIC ${STB_INCLUDE_DIR} ${MINIAUDIO_INCLUDE_DIR})
 target_compile_features   (mtmd PRIVATE cxx_std_17)
 
 if (BUILD_SHARED_LIBS)
diff --git a/tools/mtmd/mtmd-helper.cpp b/tools/mtmd/mtmd-helper.cpp
index 4094074..cd92a73 100644
--- a/tools/mtmd/mtmd-helper.cpp
+++ b/tools/mtmd/mtmd-helper.cpp
@@ -27,10 +27,10 @@
 #define MA_NO_ENGINE
 #define MA_NO_GENERATION
 #define MA_API static
-#include "miniaudio/miniaudio.h"
+#include <miniaudio.h>
 
 #define STB_IMAGE_IMPLEMENTATION
-#include "stb/stb_image.h"
+#include <stb_image.h>
 
 #ifdef MTMD_INTERNAL_HEADER
 #error "mtmd-helper is a public library outside of mtmd. it must not include internal headers"
diff --git a/tools/server/CMakeLists.txt b/tools/server/CMakeLists.txt
index 7d42743..c47a1e3 100644
--- a/tools/server/CMakeLists.txt
+++ b/tools/server/CMakeLists.txt
@@ -42,7 +42,9 @@ set_target_properties(${TARGET} PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
 
 target_include_directories(${TARGET} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 target_include_directories(${TARGET} PRIVATE ../mtmd ${CMAKE_SOURCE_DIR})
-target_link_libraries(${TARGET} PUBLIC server-context llama-ui cpp-httplib ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PUBLIC server-context llama-ui ${CMAKE_THREAD_LIBS_INIT})
+find_package(httplib CONFIG REQUIRED GLOBAL)
+target_link_libraries(${TARGET} PRIVATE httplib::httplib)
 
 if(LLAMA_TOOLS_INSTALL)
     install(TARGETS ${TARGET} LIBRARY)
diff --git a/tools/server/server-http.cpp b/tools/server/server-http.cpp
index c0a9af8..69aa240 100644
--- a/tools/server/server-http.cpp
+++ b/tools/server/server-http.cpp
@@ -3,7 +3,7 @@
 #include "server-common.h"
 #include "ui.h"
 
-#include <cpp-httplib/httplib.h>
+#include <httplib.h>
 
 #include <cstdlib>
 #include <functional>
diff --git a/tools/server/server-models.cpp b/tools/server/server-models.cpp
index 47b6c2a..60be060 100644
--- a/tools/server/server-models.cpp
+++ b/tools/server/server-models.cpp
@@ -5,7 +5,7 @@
 #include "preset.h"
 #include "download.h"
 
-#include <cpp-httplib/httplib.h> // TODO: remove this once we use HTTP client from download.h
+#include <httplib.h> // TODO: remove this once we use HTTP client from download.h
 #include <sheredom/subprocess.h>
 
 #include <functional>
diff --git a/tools/tts/CMakeLists.txt b/tools/tts/CMakeLists.txt
index 26a8bb8..dfc7cb1 100644
--- a/tools/tts/CMakeLists.txt
+++ b/tools/tts/CMakeLists.txt
@@ -6,3 +6,6 @@ target_compile_features(${TARGET} PRIVATE cxx_std_17)
 if(LLAMA_TOOLS_INSTALL)
     install(TARGETS ${TARGET} RUNTIME)
 endif()
+
+find_package(nlohmann_json CONFIG REQUIRED GLOBAL)
+target_link_libraries(${TARGET} PRIVATE nlohmann_json::nlohmann_json)
