diff --git a/common/CMakeLists.txt b/common/CMakeLists.txt
index 17146ff..c53e1b9 100644
--- a/common/CMakeLists.txt
+++ b/common/CMakeLists.txt
@@ -63,7 +63,7 @@ add_library(${TARGET} STATIC
     console.cpp
     console.h
     json-schema-to-grammar.cpp
-    json.hpp
+    # json.hpp
     llguidance.cpp
     log.cpp
     log.h
@@ -82,6 +82,8 @@ if (BUILD_SHARED_LIBS)
 endif()
 
 set(LLAMA_COMMON_EXTRA_LIBS build_info)
+find_package(nlohmann_json CONFIG REQUIRED)
+target_link_libraries(${TARGET} PRIVATE nlohmann_json::nlohmann_json)
 
 # Use curl to download model url
 if (LLAMA_CURL)
diff --git a/common/common.cpp b/common/common.cpp
index 18ffb4e..2d0b6b1 100644
--- a/common/common.cpp
+++ b/common/common.cpp
@@ -9,7 +9,7 @@
 #include "log.h"
 // Change JSON_ASSERT from assert() to GGML_ASSERT:
 #define JSON_ASSERT GGML_ASSERT
-#include "json.hpp"
+#include <nlohmann/json.hpp>
 #include "llama.h"
 
 #include <algorithm>
diff --git a/common/json-schema-to-grammar.h b/common/json-schema-to-grammar.h
index 4613f5d..6763e6d 100644
--- a/common/json-schema-to-grammar.h
+++ b/common/json-schema-to-grammar.h
@@ -3,7 +3,7 @@
 #include "ggml.h"
 // Change JSON_ASSERT from assert() to GGML_ASSERT:
 #define JSON_ASSERT GGML_ASSERT
-#include "json.hpp"
+#include <nlohmann/json.hpp>
 
 std::string json_schema_to_grammar(const nlohmann::ordered_json & schema,
                                    bool force_gbnf = false);
diff --git a/common/minja/chat-template.hpp b/common/minja/chat-template.hpp
index 882ba41..6ed9d4d 100644
--- a/common/minja/chat-template.hpp
+++ b/common/minja/chat-template.hpp
@@ -9,7 +9,7 @@
 #pragma once
 
 #include "minja.hpp"
-#include <json.hpp>
+#include <nlohmann/json.hpp>
 #include <string>
 #include <vector>
 
diff --git a/common/minja/minja.hpp b/common/minja/minja.hpp
index fa4c34d..47aa4ac 100644
--- a/common/minja/minja.hpp
+++ b/common/minja/minja.hpp
@@ -16,7 +16,7 @@
 #include <stdexcept>
 #include <sstream>
 #include <unordered_set>
-#include <json.hpp>
+#include <nlohmann/json.hpp>
 
 using json = nlohmann::ordered_json;
 
diff --git a/examples/server/CMakeLists.txt b/examples/server/CMakeLists.txt
index aee9038..294a18b 100644
--- a/examples/server/CMakeLists.txt
+++ b/examples/server/CMakeLists.txt
@@ -12,7 +12,7 @@ endif()
 set(TARGET_SRCS
     server.cpp
     utils.hpp
-    httplib.h
+    # httplib.h
 )
 set(PUBLIC_ASSETS
     index.html.gz
@@ -37,6 +37,9 @@ install(TARGETS ${TARGET} RUNTIME)
 target_include_directories(${TARGET} PRIVATE ${CMAKE_SOURCE_DIR})
 target_link_libraries(${TARGET} PRIVATE common ${CMAKE_THREAD_LIBS_INIT})
 
+find_package(httplib CONFIG REQUIRED)
+target_link_libraries(${TARGET} PRIVATE httplib::httplib)
+
 if (LLAMA_SERVER_SSL)
     find_package(OpenSSL REQUIRED)
     target_link_libraries(${TARGET} PRIVATE OpenSSL::SSL OpenSSL::Crypto)
diff --git a/examples/server/server.cpp b/examples/server/server.cpp
index c2f1afe..d9625ee 100644
--- a/examples/server/server.cpp
+++ b/examples/server/server.cpp
@@ -10,7 +10,7 @@
 
 // Change JSON_ASSERT from assert() to GGML_ASSERT:
 #define JSON_ASSERT GGML_ASSERT
-#include "json.hpp"
+#include <nlohmann/json.hpp>
 // mime type for sending response
 #define MIMETYPE_JSON "application/json; charset=utf-8"
 
diff --git a/examples/server/utils.hpp b/examples/server/utils.hpp
index 58cdd6a..2898e62 100644
--- a/examples/server/utils.hpp
+++ b/examples/server/utils.hpp
@@ -9,11 +9,11 @@
 #define CPPHTTPLIB_FORM_URL_ENCODED_PAYLOAD_MAX_LENGTH 1048576
 // disable Nagle's algorithm
 #define CPPHTTPLIB_TCP_NODELAY true
-#include "httplib.h"
+#include <httplib.h>
 
 // Change JSON_ASSERT from assert() to GGML_ASSERT:
 #define JSON_ASSERT GGML_ASSERT
-#include "json.hpp"
+#include <nlohmann/json.hpp>
 #include "chat.h"
 
 #include <random>
