diff --git a/CMakeLists.txt b/CMakeLists.txt
index 74b48d2..172c5d9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -169,6 +169,9 @@ if (LLAMA_BUILD_COMMON AND LLAMA_BUILD_TESTS AND NOT CMAKE_JS_VERSION)
     add_subdirectory(tests)
 endif()
 
+if (LLAMA_BUILD_SERVER)
+    add_subdirectory(examples/server)
+endif()
 if (LLAMA_BUILD_COMMON AND LLAMA_BUILD_EXAMPLES)
     add_subdirectory(examples)
     add_subdirectory(pocs)
diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index 66cfab2..9df5985 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -41,9 +41,6 @@ else()
     add_subdirectory(perplexity)
     add_subdirectory(quantize)
     add_subdirectory(retrieval)
-    if (LLAMA_BUILD_SERVER)
-        add_subdirectory(server)
-    endif()
     add_subdirectory(save-load-state)
     add_subdirectory(run)
     add_subdirectory(simple)
