diff --git a/CMakeLists.txt b/CMakeLists.txt
index a3c9606..6f570a4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -569,7 +569,7 @@ target_link_libraries( lapacktest
 
 # ----------------------------------------
 # compile tester library
-add_library( tester ${libtest_all} )
+add_library( tester EXCLUDE_FROM_ALL ${libtest_all} )
 target_link_libraries( tester
     magma
     lapacktest
@@ -655,11 +655,12 @@ endif()
 
 set( CMAKE_RUNTIME_OUTPUT_DIRECTORY "${SPARSE_TEST_DIR}" )
 cmake_policy( SET CMP0037 NEW)
+set(sparse_testing_all "")
 foreach( TEST ${sparse_testing_all} )
     string( REGEX REPLACE "\\.(cpp|f90|F90)"     "" EXE ${TEST} )
     string( REGEX REPLACE "${SPARSE_TEST_DIR}/" "" EXE ${EXE} )
     #message( "${TEST} --> ${EXE}" )
-    add_executable( ${EXE} ${TEST} )
+    add_executable( ${EXE} EXCLUDE_FROM_ALL ${TEST} )
     target_link_libraries( ${EXE} magma_sparse magma )
     list( APPEND sparse-testing ${EXE} )
 endforeach()
