diff --git a/Package.swift b/Package.swift
index a02b8ed..5ee71f3 100644
--- a/Package.swift
+++ b/Package.swift
@@ -52,9 +52,7 @@ let package = Package(
         .library(name: "NIOSSL", targets: ["NIOSSL"]),
         .executable(name: "NIOTLSServer", targets: ["NIOTLSServer"]),
         .executable(name: "NIOSSLHTTP1Client", targets: ["NIOSSLHTTP1Client"]),
-/* This target is used only for symbol mangling. It's added and removed automatically because it emits build warnings. MANGLE_START
-        .library(name: "CNIOBoringSSL", type: .static, targets: ["CNIOBoringSSL"]),
-MANGLE_END */
+        .library(name: "CNIOBoringSSL", type: .dynamic, targets: ["CNIOBoringSSL"]),
     ],
     dependencies: generateDependencies(),
     targets: [
