diff --git a/CMakeLists.txt b/CMakeLists.txt
index 096f7ec..caf5fa0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -63,3 +63,14 @@ install(
     DIRECTORY ${PROJECT_SOURCE_DIR}/include/
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 )
+
+include(CMakePackageConfigHelpers)
+get_filename_component(CONFIG_FILE_PATH ${CMAKE_CURRENT_BINARY_DIR}/cudnn-frontend-config.cmake ABSOLUTE)
+configure_package_config_file(cmake/config-template.cmake ${CONFIG_FILE_PATH} INSTALL_DESTINATION lib/cmake)
+install(FILES ${CONFIG_FILE_PATH} DESTINATION lib/cmake)
+
+install(TARGETS cudnn_frontend EXPORT cudnn-frontend-targets)
+install(EXPORT cudnn-frontend-targets
+    FILE cudnn-frontend-targets.cmake # NAMESPACE unofficial::
+    DESTINATION lib/cmake
+)
\ No newline at end of file
